// @sfxcode/formkit-primevue/dist/formkit-primevue.es.js
import { defineComponent as x, getCurrentInstance as at, watch as ye, watchEffect as ae, reactive as yn, onUnmounted as vn, h as Ae, ref as Me, inject as Oe, onMounted as hn, computed as M, provide as _n, toRef as gn, onBeforeUnmount as $n, markRaw as wn, isRef as kn, createTextVNode as Cn, resolveComponent as E, openBlock as A, createElementBlock as O, createVNode as T, unref as a, toDisplayString as ce, createCommentVNode as oe, normalizeStyle as pe, normalizeClass as U, createElementVNode as Bt, Fragment as Sn, renderList as Vn } from "vue";
const zt = [
  "__key",
  "__init",
  "__shim",
  "__original",
  "__index",
  "__prevKey"
];
function Je() {
  return Math.random().toString(36).substring(2, 15);
}
function In(e, i) {
  const t = e instanceof Set ? e : new Set(e);
  return i && i.forEach((n) => t.add(n)), [...t];
}
function j(e, i) {
  return Object.prototype.hasOwnProperty.call(e, i);
}
function ue(e, i, t = !0, n = ["__key"]) {
  if (e === i)
    return !0;
  if (typeof i == "object" && typeof e == "object") {
    if (e instanceof Map || e instanceof Set || e instanceof Date || e === null || i === null || Object.keys(e).length !== Object.keys(i).length)
      return !1;
    for (const l of n)
      if ((l in e || l in i) && e[l] !== i[l])
        return !1;
    for (const l in e)
      if (!(l in i) || e[l] !== i[l] && !t || t && !ue(e[l], i[l], t, n))
        return !1;
    return !0;
  }
  return !1;
}
function kt(e) {
  const i = typeof e;
  if (i === "number")
    return !1;
  if (e === void 0)
    return !0;
  if (i === "string")
    return e === "";
  if (i === "object") {
    if (e === null)
      return !0;
    for (const t in e)
      return !1;
    return !(e instanceof RegExp || e instanceof Date);
  }
  return !1;
}
function st(e) {
  return Object.prototype.toString.call(e) === "[object Object]";
}
function dt(e) {
  return st(e) || Array.isArray(e);
}
function _e(e) {
  if (st(e) === !1 || e.__FKNode__ || e.__POJO__ === !1)
    return !1;
  const i = e.constructor;
  if (i === void 0)
    return !0;
  const t = i.prototype;
  return !(st(t) === !1 || t.hasOwnProperty("isPrototypeOf") === !1);
}
function Ye(e, i, t = !1, n = !1) {
  if (i === null)
    return null;
  const l = {};
  if (typeof i == "string")
    return i;
  for (const s in e)
    if (j(i, s) && (i[s] !== void 0 || !n)) {
      if (t && Array.isArray(e[s]) && Array.isArray(i[s])) {
        l[s] = e[s].concat(i[s]);
        continue;
      }
      if (i[s] === void 0)
        continue;
      _e(e[s]) && _e(i[s]) ? l[s] = Ye(e[s], i[s], t, n) : l[s] = i[s];
    } else
      l[s] = e[s];
  for (const s in i)
    !j(l, s) && i[s] !== void 0 && (l[s] = i[s]);
  return l;
}
function Ln(e) {
  if (e[0] !== '"' && e[0] !== "'" || e[0] !== e[e.length - 1])
    return !1;
  const i = e[0];
  for (let t = 1; t < e.length; t++)
    if (e[t] === i && (t === 1 || e[t - 1] !== "\\") && t !== e.length - 1)
      return !1;
  return !0;
}
function An(e) {
  if (!e.length)
    return "";
  let i = "", t = "";
  for (let n = 0; n < e.length; n++) {
    const l = e.charAt(n);
    (l !== "\\" || t === "\\") && (i += l), t = l;
  }
  return i;
}
function me(...e) {
  return e.reduce((i, t) => {
    const { value: n, name: l, modelValue: s, config: r, plugins: o, ...u } = t;
    return Object.assign(i, u);
  }, {});
}
function On(e) {
  const i = [];
  let t = "", n = 0, l = "", s = "";
  for (let r = 0; r < e.length; r++) {
    const o = e.charAt(r);
    o === l && s !== "\\" ? l = "" : (o === "'" || o === '"') && !l && s !== "\\" ? l = o : o === "(" && !l ? n++ : o === ")" && !l && n--, o === "," && !l && n === 0 ? (i.push(t), t = "") : (o !== " " || l) && (t += o), s = o;
  }
  return t && i.push(t), i;
}
function Ct(e, i) {
  const t = {}, n = i.filter((s) => s instanceof RegExp), l = new Set(i);
  for (const s in e)
    !l.has(s) && !n.some((r) => r.test(s)) && (t[s] = e[s]);
  return t;
}
function St(e, i) {
  const t = {}, n = i.filter((l) => l instanceof RegExp);
  return i.forEach((l) => {
    l instanceof RegExp || (t[l] = e[l]);
  }), Object.keys(e).forEach((l) => {
    n.some((s) => s.test(l)) && (t[l] = e[l]);
  }), t;
}
function Fe(e) {
  return e.replace(/-([a-z0-9])/gi, (i, t) => t.toUpperCase());
}
function Ut(e) {
  return e.replace(/([a-z0-9])([A-Z])/g, (i, t, n) => t + "-" + n.toLowerCase()).replace(" ", "-").toLowerCase();
}
function Pn(e, i = zt) {
  if (e !== null && typeof e == "object") {
    let t;
    if (Array.isArray(e) ? t = [...e] : _e(e) && (t = { ...e }), t)
      return Mn(e, t, i), t;
  }
  return e;
}
function Pe(e, i = zt) {
  if (e === null || e instanceof RegExp || e instanceof Date || e instanceof Map || e instanceof Set || typeof File == "function" && e instanceof File)
    return e;
  let t;
  Array.isArray(e) ? t = e.map((n) => typeof n == "object" ? Pe(n, i) : n) : t = Object.keys(e).reduce((n, l) => (n[l] = typeof e[l] == "object" ? Pe(e[l], i) : e[l], n), {});
  for (const n of i)
    n in e && Object.defineProperty(t, n, {
      enumerable: !1,
      value: e[n]
    });
  return t;
}
function de(e) {
  return typeof e == "object" ? Pe(e) : e;
}
function xn(e, i) {
  if (!e || typeof e != "object")
    return null;
  const t = i.split(".");
  let n = e;
  for (const l in t) {
    const s = t[l];
    if (j(n, s) && (n = n[s]), +l === t.length - 1)
      return n;
    if (!n || typeof n != "object")
      return null;
  }
  return null;
}
function Y(e) {
  return e !== void 0 && e !== "false" && e !== !1 ? !0 : void 0;
}
function xe(e) {
  return Object.isFrozen(e) ? e : Object.defineProperty(e, "__init", {
    enumerable: !1,
    value: !0
  });
}
function pt(e) {
  return e.normalize("NFD").replace(/[\u0300-\u036f]/g, "").toLowerCase().replace(/[^a-z0-9]/g, " ").trim().replace(/\s+/g, "-");
}
function Mn(e, i, t) {
  for (const n of t)
    n in e && Object.defineProperty(i, n, {
      enumerable: !1,
      value: e[n]
    });
  return i;
}
function jn(e) {
  let i = !1;
  return (...t) => {
    if (!i)
      return i = !0, queueMicrotask(() => i = !1), e(...t);
  };
}
function mt() {
  const e = [];
  let i = 0;
  const t = (l) => e.push(l), n = (l) => {
    const s = e[i];
    return typeof s == "function" ? s(l, (r) => (i++, n(r === void 0 ? l : r))) : (i = 0, l);
  };
  return t.dispatch = n, t.unshift = (l) => e.unshift(l), t.remove = (l) => {
    const s = e.indexOf(l);
    s > -1 && e.splice(s, 1);
  }, t;
}
function Kt() {
  const e = /* @__PURE__ */ new Map(), i = /* @__PURE__ */ new Map();
  let t;
  const n = (l, s) => {
    if (t) {
      t.set(s.name, [l, s]);
      return;
    }
    e.has(s.name) && e.get(s.name).forEach((r) => {
      (s.origin === l || r.modifiers.includes("deep")) && r.listener(s);
    }), s.bubble && l.bubble(s);
  };
  return n.flush = () => {
    e.clear(), i.clear(), t == null || t.clear();
  }, n.on = (l, s) => {
    const [r, ...o] = l.split("."), u = s.receipt || Je(), d = {
      modifiers: o,
      event: r,
      listener: s,
      receipt: u
    };
    return e.has(r) ? e.get(r).push(d) : e.set(r, [d]), i.has(u) ? i.get(u).push(r) : i.set(u, [r]), u;
  }, n.off = (l) => {
    var s;
    i.has(l) && ((s = i.get(l)) === null || s === void 0 || s.forEach((r) => {
      const o = e.get(r);
      Array.isArray(o) && e.set(r, o.filter((u) => u.receipt !== l));
    }), i.delete(l));
  }, n.pause = (l) => {
    t || (t = /* @__PURE__ */ new Map()), l && l.walk((s) => s._e.pause());
  }, n.play = (l) => {
    if (!t)
      return;
    const s = t;
    t = void 0, s.forEach(([r, o]) => n(r, o)), l && l.walk((r) => r._e.play());
  }, n;
}
function En(e, i, t, n, l = !0) {
  return i._e(e, {
    payload: n,
    name: t,
    bubble: l,
    origin: e
  }), e;
}
function Rn(e, i, t) {
  return Qe(e.parent) && e.parent._e(e.parent, t), e;
}
function Fn(e, i, t, n) {
  return i._e.on(t, n);
}
function Tn(e, i, t) {
  return i._e.off(t), e;
}
const bt = mt();
bt((e, i) => (e.message || (e.message = `E${e.code}`), i(e)));
const yt = mt();
yt((e, i) => {
  e.message || (e.message = `W${e.code}`);
  const t = i(e);
  return console && typeof console.warn == "function" && console.warn(t.message), t;
});
function je(e, i = {}) {
  yt.dispatch({ code: e, data: i });
}
function W(e, i = {}) {
  throw Error(bt.dispatch({ code: e, data: i }).message);
}
function re(e, i) {
  const t = {
    blocking: !1,
    key: Je(),
    meta: {},
    type: "state",
    visible: !0,
    ...e
  };
  return i && t.value && t.meta.localize !== !1 && (t.value = i.t(t), t.meta.locale = i.config.locale), t;
}
const Vt = {
  apply: qn,
  set: Bn,
  remove: qt,
  filter: Un,
  reduce: Kn,
  release: Hn,
  touch: zn
};
function Dn(e = !1) {
  const i = {};
  let t, n = e, l = [];
  const s = /* @__PURE__ */ new Map();
  let r;
  const o = new Proxy(i, {
    get(...u) {
      const [d, _] = u;
      return _ === "buffer" ? n : _ === "_b" ? l : _ === "_m" ? s : _ === "_r" ? r : j(Vt, _) ? Vt[_].bind(null, i, o, t) : Reflect.get(...u);
    },
    set(u, d, _) {
      return d === "_n" ? (t = _, r === "__n" && Wt(t, o), !0) : d === "_b" ? (l = _, !0) : d === "buffer" ? (n = _, !0) : d === "_r" ? (r = _, !0) : (W(101, t), !1);
    }
  });
  return o;
}
function Bn(e, i, t, n) {
  if (i.buffer)
    return i._b.push([[n]]), i;
  if (e[n.key] !== n) {
    if (typeof n.value == "string" && n.meta.localize !== !1) {
      const s = n.value;
      n.value = t.t(n), n.value !== s && (n.meta.locale = t.props.locale);
    }
    const l = `message-${j(e, n.key) ? "updated" : "added"}`;
    e[n.key] = Object.freeze(t.hook.message.dispatch(n)), t.emit(l, n);
  }
  return i;
}
function zn(e, i) {
  for (const t in e) {
    const n = { ...e[t] };
    i.set(n);
  }
}
function qt(e, i, t, n) {
  if (j(e, n)) {
    const l = e[n];
    delete e[n], t.emit("message-removed", l);
  }
  return i.buffer === !0 && (i._b = i._b.filter((l) => (l[0] = l[0].filter((s) => s.key !== n), l[1] || l[0].length))), i;
}
function Un(e, i, t, n, l) {
  for (const s in e) {
    const r = e[s];
    (!l || r.type === l) && !n(r) && qt(e, i, t, s);
  }
}
function Kn(e, i, t, n, l) {
  for (const s in e) {
    const r = e[s];
    l = n(l, r);
  }
  return l;
}
function qn(e, i, t, n, l) {
  if (Array.isArray(n)) {
    if (i.buffer) {
      i._b.push([n, l]);
      return;
    }
    const s = new Set(n.map((r) => (i.set(r), r.key)));
    typeof l == "string" ? i.filter((r) => r.type !== l || s.has(r.key)) : typeof l == "function" && i.filter((r) => !l(r) || s.has(r.key));
  } else
    for (const s in n) {
      const r = t.at(s);
      r ? r.store.apply(n[s], l) : Nn(t, i, s, n[s], l);
    }
}
function Wn(e, ...i) {
  const t = `${e.name}-set`, n = (l) => re({
    key: pt(l),
    type: "error",
    value: l,
    meta: { source: t, autoClear: !0 }
  });
  return i.filter((l) => !!l).map((l) => {
    if (typeof l == "string" && (l = [l]), Array.isArray(l))
      return l.map((s) => n(s));
    {
      const s = {};
      for (const r in l)
        Array.isArray(l[r]) ? s[r] = l[r].map((o) => n(o)) : s[r] = [n(l[r])];
      return s;
    }
  });
}
function Nn(e, i, t, n, l) {
  var s;
  const r = i._m;
  r.has(t) || r.set(t, []), i._r || (i._r = Wt(e, i)), (s = r.get(t)) === null || s === void 0 || s.push([n, l]);
}
function Wt(e, i) {
  return e.on("child.deep", ({ payload: t }) => {
    i._m.forEach((n, l) => {
      e.at(l) === t && (n.forEach(([s, r]) => {
        t.store.apply(s, r);
      }), i._m.delete(l));
    }), i._m.size === 0 && i._r && (e.off(i._r), i._r = void 0);
  });
}
function Hn(e, i) {
  i.buffer = !1, i._b.forEach(([t, n]) => i.apply(t, n)), i._b = [];
}
function Zn() {
  const e = {};
  let i;
  return {
    count: (...t) => Jn(i, e, ...t),
    init(t) {
      i = t, t.on("message-added.deep", It(e, 1)), t.on("message-removed.deep", It(e, -1));
    },
    merge: (t) => Lt(i, e, t),
    settled(t) {
      return j(e, t) ? e[t].promise : Promise.resolve();
    },
    unmerge: (t) => Lt(i, e, t, !0),
    value(t) {
      return j(e, t) ? e[t].count : 0;
    }
  };
}
function Jn(e, i, t, n, l = 0) {
  if (n = Yn(n || t), !j(i, t)) {
    const s = {
      condition: n,
      count: 0,
      name: t,
      node: e,
      promise: Promise.resolve(),
      resolve: () => {
      }
      // eslint-disable-line @typescript-eslint/no-empty-function
    };
    i[t] = s, l = e.store.reduce((r, o) => r + s.condition(o) * 1, l), e.each((r) => {
      r.ledger.count(s.name, s.condition), l += r.ledger.value(s.name);
    });
  }
  return Nt(i[t], l).promise;
}
function Yn(e) {
  return typeof e == "function" ? e : (i) => i.type === e;
}
function Nt(e, i) {
  const t = e.count, n = e.count + i;
  return e.count = n, t === 0 && n !== 0 ? (e.node.emit(`unsettled:${e.name}`, e.count, !1), e.promise = new Promise((l) => e.resolve = l)) : t !== 0 && n === 0 && (e.node.emit(`settled:${e.name}`, e.count, !1), e.resolve()), e.node.emit(`count:${e.name}`, e.count, !1), e;
}
function It(e, i) {
  return (t) => {
    for (const n in e) {
      const l = e[n];
      l.condition(t.payload) && Nt(l, i);
    }
  };
}
function Lt(e, i, t, n = !1) {
  for (const l in i) {
    const s = i[l].condition;
    n || t.ledger.count(l, s);
    const r = t.ledger.value(l) * (n ? -1 : 1);
    if (e)
      do
        e.ledger.count(l, s, r), e = e.parent;
      while (e);
  }
}
const vt = /* @__PURE__ */ new Map(), Te = /* @__PURE__ */ new Map(), ht = Kt(), Qn = [];
function Xn(e) {
  e.props.id && (vt.set(e.props.id, e), Te.set(e, e.props.id), ht(e, {
    payload: e,
    name: e.props.id,
    bubble: !1,
    origin: e
  }));
}
function Gn(e) {
  if (Te.has(e)) {
    const i = Te.get(e);
    Te.delete(e), vt.delete(i), ht(e, {
      payload: null,
      name: i,
      bubble: !1,
      origin: e
    });
  }
}
function Ht(e) {
  return vt.get(e);
}
function ei(e, i) {
  Qn.push(ht.on(e, i));
}
function At(e, i, t) {
  let n = !0;
  return i in e.config._t ? n = !1 : e.emit(`config:${i}`, t, !1), i in e.props || (e.emit("prop", { prop: i, value: t }), e.emit(`prop:${i}`, t)), n;
}
function ti(e) {
  const i = document.getElementById(e);
  if (i instanceof HTMLFormElement) {
    const t = new Event("submit", { cancelable: !0, bubbles: !0 });
    i.dispatchEvent(t);
    return;
  }
  je(151, e);
}
function ni(e) {
  const i = (t) => {
    for (const n in t.store) {
      const l = t.store[n];
      l.type === "error" || l.type === "ui" && n === "incomplete" ? t.store.remove(n) : l.type === "state" && t.store.set({ ...l, value: !1 });
    }
  };
  i(e), e.walk(i);
}
function ii(e, i) {
  const t = typeof e == "string" ? Ht(e) : e;
  if (t) {
    const n = (s) => de(s.props.initial) || (s.type === "group" ? {} : s.type === "list" ? [] : void 0);
    t._e.pause(t);
    const l = de(i);
    return i && !kt(i) && (t.props.initial = dt(l) ? xe(l) : l), t.input(n(t), !1), t.walk((s) => s.input(n(s), !1)), t.input(kt(l) && l ? l : n(t), !1), t._e.play(t), ni(t), t.emit("reset", t), t;
  }
  je(152, e);
}
const li = {
  delimiter: ".",
  delay: 0,
  locale: "en",
  rootClasses: (e) => ({ [`formkit-${Ut(e)}`]: !0 })
}, Zt = Symbol("index"), rt = Symbol("removed"), ot = Symbol("moved"), Jt = Symbol("inserted");
function ai(e) {
  return e.type === "list" && Array.isArray(e._value);
}
function Qe(e) {
  return e && typeof e == "object" && e.__FKNode__ === !0;
}
const De = (e, i, t) => {
  W(102, [e, t]);
}, si = {
  _c: S(Ii, De, !1),
  add: S(_i),
  addProps: S(hi),
  address: S(Ai, De, !1),
  at: S(Oi),
  bubble: S(Rn),
  clearErrors: S(Ti),
  calm: S(bi),
  config: S(!1),
  define: S(vi),
  disturb: S(mi),
  destroy: S(yi),
  extend: S(zi),
  hydrate: S(di),
  index: S(Vi, Si, !1),
  input: S(Xt),
  each: S(wi),
  emit: S(En),
  find: S(xi),
  on: S(Fn),
  off: S(Tn),
  parent: S(!1, gi),
  plugins: S(!1),
  remove: S($i),
  root: S(ji, De, !1),
  reset: S(Fi),
  resetConfig: S(Ci),
  setErrors: S(tn),
  submit: S(Ri),
  t: S(Ei),
  use: S(_t),
  name: S(Li, !1, !1),
  walk: S(ki)
};
function ri() {
  return new Map(Object.entries(si));
}
function S(e, i, t = !0) {
  return {
    get: e ? (n, l) => t ? (...s) => e(n, l, ...s) : e(n, l) : !1,
    set: i !== void 0 ? i : De.bind(null)
  };
}
function oi() {
  const e = /* @__PURE__ */ new Map();
  return new Proxy(e, {
    get(i, t) {
      return e.has(t) || e.set(t, mt()), e.get(t);
    }
  });
}
let Yt = 0, ui = 0;
function fi(e) {
  var i, t;
  return ((i = e.parent) === null || i === void 0 ? void 0 : i.type) === "list" ? Zt : e.name || `${((t = e.props) === null || t === void 0 ? void 0 : t.type) || "input"}_${++Yt}`;
}
function Qt(e) {
  return e.type === "group" ? xe(e.value && typeof e.value == "object" && !Array.isArray(e.value) ? e.value : {}) : e.type === "list" ? xe(Array.isArray(e.value) ? e.value : []) : e.value;
}
function Xt(e, i, t, n = !0) {
  return i._value = ci(e, e.hook.input.dispatch(t)), e.emit("input", i._value), e.isCreated && e.type === "input" && ue(i._value, i.value) ? (e.emit("commitRaw", i.value), i.settled) : (i.isSettled && e.disturb(), n ? (i._tmo && clearTimeout(i._tmo), i._tmo = setTimeout(Ue, e.props.delay, e, i)) : Ue(e, i), i.settled);
}
function ci(e, i) {
  switch (e.type) {
    case "input":
      break;
    case "group":
      (!i || typeof i != "object") && W(107, [e, i]);
      break;
    case "list":
      Array.isArray(i) || W(108, [e, i]);
      break;
  }
  return i;
}
function Ue(e, i, t = !0, n = !0) {
  i._value = i.value = e.hook.commit.dispatch(i._value), e.type !== "input" && n && e.hydrate(), e.emit("commitRaw", i.value), e.emit("commit", i.value), t && e.calm();
}
function Gt(e, { name: i, value: t, from: n }) {
  if (!Object.isFrozen(e._value)) {
    if (ai(e)) {
      const l = t === rt ? [] : t === ot && typeof n == "number" ? e._value.splice(n, 1) : [t];
      e._value.splice(i, t === ot || n === Jt ? 0 : 1, ...l);
      return;
    }
    t !== rt ? e._value[i] = t : delete e._value[i];
  }
}
function di(e, i) {
  const t = i._value;
  return e.type === "list" && e.sync && pi(e, i), i.children.forEach((n) => {
    if (typeof t == "object")
      if (n.name in t) {
        const l = n.type !== "input" || t[n.name] && typeof t[n.name] == "object" ? xe(t[n.name]) : t[n.name];
        if (!n.isSettled || !dt(l) && ue(l, n._value))
          return;
        n.input(l, !1);
      } else
        (e.type !== "list" || typeof n.name == "number") && Gt(i, { name: n.name, value: n.value }), t.__init || (n.type === "group" ? n.input({}, !1) : n.type === "list" ? n.input([], !1) : n.input(void 0, !1));
  }), e;
}
function pi(e, i) {
  const t = e._value;
  if (!Array.isArray(t))
    return;
  const n = [], l = new Set(i.children), s = /* @__PURE__ */ new Map();
  t.forEach((o, u) => {
    if (i.children[u] && i.children[u]._value === o)
      n.push(i.children[u]), l.delete(i.children[u]);
    else {
      n.push(null);
      const d = s.get(o) || [];
      d.push(u), s.set(o, d);
    }
  }), l.size && s.size && l.forEach((o) => {
    if (s.has(o._value)) {
      const u = s.get(o._value), d = u.shift();
      n[d] = o, l.delete(o), u.length || s.delete(o._value);
    }
  });
  const r = [];
  for (s.forEach((o) => {
    r.push(...o);
  }); l.size && r.length; ) {
    const o = l.values().next().value, u = r.shift();
    if (u === void 0)
      break;
    n[u] = o, l.delete(o);
  }
  r.forEach((o, u) => {
    n[o] = Wi({ value: u });
  }), l.size && l.forEach((o) => {
    if (!("__FKP" in o)) {
      const u = o._c.parent;
      if (!u || Ni(u))
        return;
      u.ledger.unmerge(o), o._c.parent = null, o.destroy();
    }
  }), i.children = n;
}
function mi(e, i) {
  var t;
  return i._d <= 0 && (i.isSettled = !1, e.emit("settled", !1, !1), i.settled = new Promise((n) => {
    i._resolve = n;
  }), e.parent && ((t = e.parent) === null || t === void 0 || t.disturb())), i._d++, e;
}
function bi(e, i, t) {
  var n;
  if (t !== void 0 && e.type !== "input")
    return Gt(i, t), Ue(e, i, !0, !1);
  i._d > 0 && i._d--, i._d === 0 && (i.isSettled = !0, e.emit("settled", !0, !1), e.parent && ((n = e.parent) === null || n === void 0 || n.calm({ name: e.name, value: i.value })), i._resolve && i._resolve(i.value));
}
function yi(e, i) {
  e.emit("destroying", e), e.store.filter(() => !1), e.parent && (e.parent.emit("childRemoved", e), e.parent.remove(e)), Gn(e), e.emit("destroyed", e), i._e.flush(), i._value = i.value = void 0;
  for (const t in i.context)
    delete i.context[t];
  i.plugins.clear(), i.context = null;
}
function vi(e, i, t) {
  i.type = t.type, i.props.definition = Pe(t), i.value = i._value = Qt({
    type: e.type,
    value: i.value
  }), t.forceTypeProp && (e.props.type && (e.props.originalType = e.props.type), i.props.type = t.forceTypeProp), t.family && (i.props.family = t.family), t.features && t.features.forEach((n) => n(e)), t.props && e.addProps(t.props), e.emit("defined", t);
}
function hi(e, i, t) {
  var n;
  if (e.props.attrs) {
    const l = { ...e.props.attrs };
    e.props._emit = !1;
    for (const r in l) {
      const o = Fe(r);
      t.includes(o) && (e.props[o] = l[r], delete l[r]);
    }
    const s = de(i._value);
    e.props.initial = e.type !== "input" ? xe(s) : s, e.props._emit = !0, e.props.attrs = l, e.props.definition && (e.props.definition.props = [
      ...((n = e.props.definition) === null || n === void 0 ? void 0 : n.props) || [],
      ...t
    ]);
  }
  return e.emit("added-props", t), e;
}
function _i(e, i, t, n) {
  if (e.type === "input" && W(100, e), t.parent && t.parent !== e && t.parent.remove(t), !i.children.includes(t)) {
    if (n !== void 0 && e.type === "list") {
      const l = i.children[n];
      l && "__FKP" in l ? (t._c.uid = l.uid, i.children.splice(n, 1, t)) : i.children.splice(n, 0, t), Array.isArray(e.value) && e.value.length < i.children.length && e.disturb().calm({
        name: n,
        value: t.value,
        from: Jt
      });
    } else
      i.children.push(t);
    t.isSettled || e.disturb();
  }
  if (t.parent !== e) {
    if (t.parent = e, t.parent !== e)
      return e.remove(t), t.parent.add(t), e;
  } else
    t.use(e.plugins);
  return Ue(e, i, !1), e.ledger.merge(t), e.emit("child", t), e;
}
function gi(e, i, t, n) {
  return Qe(n) ? (e.parent && e.parent !== n && e.parent.remove(e), i.parent = n, e.resetConfig(), n.children.includes(e) ? e.use(n.plugins) : n.add(e), !0) : n === null ? (i.parent = null, !0) : !1;
}
function $i(e, i, t) {
  const n = i.children.indexOf(t);
  if (n !== -1) {
    t.isSettled && e.disturb(), i.children.splice(n, 1);
    let l = Y(t.props.preserve), s = t.parent;
    for (; l === void 0 && s; )
      l = Y(s.props.preserve), s = s.parent;
    l ? e.calm() : e.calm({
      name: e.type === "list" ? n : t.name,
      value: rt
    }), t.parent = null, t.config._rmn = t;
  }
  return e.ledger.unmerge(t), e;
}
function wi(e, i, t) {
  i.children.forEach((n) => !("__FKP" in n) && t(n));
}
function ki(e, i, t, n = !1, l = !1) {
  i.children.some((s) => {
    if ("__FKP" in s)
      return !1;
    const r = t(s);
    return n && r === !1 ? !0 : l && r === !1 ? !1 : s.walk(t, n, l);
  });
}
function Ci(e, i) {
  const t = e.parent || void 0;
  i.config = en(e.config._t, t), e.walk((n) => n.resetConfig());
}
function _t(e, i, t, n = !0, l = !0) {
  return Array.isArray(t) || t instanceof Set ? (t.forEach((s) => _t(e, i, s)), e) : (i.plugins.has(t) || (l && typeof t.library == "function" && t.library(e), n && t(e) !== !1 && (i.plugins.add(t), e.children.forEach((s) => s.use(t)))), e);
}
function Si(e, i, t, n) {
  if (Qe(e.parent)) {
    const l = e.parent.children, s = n >= l.length ? l.length - 1 : n < 0 ? 0 : n, r = l.indexOf(e);
    return r === -1 ? !1 : (l.splice(r, 1), l.splice(s, 0, e), e.parent.children = l, e.parent.type === "list" && e.parent.disturb().calm({ name: s, value: ot, from: r }), !0);
  }
  return !1;
}
function Vi(e) {
  if (e.parent) {
    const i = [...e.parent.children].indexOf(e);
    return i === -1 ? e.parent.children.length : i;
  }
  return -1;
}
function Ii(e, i) {
  return i;
}
function Li(e, i) {
  var t;
  return ((t = e.parent) === null || t === void 0 ? void 0 : t.type) === "list" ? e.index : i.name !== Zt ? i.name : e.index;
}
function Ai(e, i) {
  return i.parent ? i.parent.address.concat([e.name]) : [e.name];
}
function Oi(e, i, t) {
  const n = typeof t == "string" ? t.split(e.config.delimiter) : t;
  if (!n.length)
    return;
  const l = n[0];
  let s = e.parent;
  for (s || (String(n[0]) === String(e.name) && n.shift(), s = e), l === "$parent" && n.shift(); s && n.length; ) {
    const r = n.shift();
    switch (r) {
      case "$root":
        s = e.root;
        break;
      case "$parent":
        s = s.parent;
        break;
      case "$self":
        s = e;
        break;
      default:
        s = s.children.find((o) => !("__FKP" in o) && String(o.name) === String(r)) || Pi(s, r);
    }
  }
  return s || void 0;
}
function Pi(e, i) {
  const t = String(i).match(/^(find)\((.*)\)$/);
  if (t) {
    const [, n, l] = t, s = l.split(",").map((r) => r.trim());
    switch (n) {
      case "find":
        return e.find(s[0], s[1]);
      default:
        return;
    }
  }
}
function xi(e, i, t, n) {
  return Mi(e, t, n);
}
function Mi(e, i, t = "name") {
  const n = typeof t == "string" ? (s) => s[t] == i : t, l = [e];
  for (; l.length; ) {
    const s = l.shift();
    if (!("__FKP" in s)) {
      if (n(s, i))
        return s;
      l.push(...s.children);
    }
  }
}
function ji(e) {
  let i = e;
  for (; i.parent; )
    i = i.parent;
  return i;
}
function en(e = {}, i) {
  let t;
  return new Proxy(e, {
    get(...n) {
      const l = n[1];
      if (l === "_t")
        return e;
      const s = Reflect.get(...n);
      if (s !== void 0)
        return s;
      if (i) {
        const r = i.config[l];
        if (r !== void 0)
          return r;
      }
      if (e.rootConfig && typeof l == "string") {
        const r = e.rootConfig[l];
        if (r !== void 0)
          return r;
      }
      return l === "delay" && (t == null ? void 0 : t.type) === "input" ? 20 : li[l];
    },
    set(...n) {
      const l = n[1], s = n[2];
      if (l === "_n")
        return t = s, e.rootConfig && e.rootConfig._add(t), !0;
      if (l === "_rmn")
        return e.rootConfig && e.rootConfig._rm(t), t = void 0, !0;
      if (!ue(e[l], s, !1)) {
        const r = Reflect.set(...n);
        return t && (t.emit(`config:${l}`, s, !1), At(t, l, s), t.walk((o) => At(o, l, s), !1, !0)), r;
      }
      return !0;
    }
  });
}
function Ei(e, i, t, n = "ui") {
  const l = typeof t == "string" ? { key: t, value: t, type: n } : t, s = e.hook.text.dispatch(l);
  return e.emit("text", s, !1), s.value;
}
function Ri(e) {
  const i = e.name;
  do {
    if (e.props.isForm === !0)
      break;
    e.parent || W(106, i), e = e.parent;
  } while (e);
  e.props.id && ti(e.props.id);
}
function Fi(e, i, t) {
  return ii(e, t);
}
function tn(e, i, t, n) {
  const l = `${e.name}-set`, s = e.hook.setErrors.dispatch({ localErrors: t, childErrors: n });
  return Wn(e, s.localErrors, s.childErrors).forEach((r) => {
    e.store.apply(r, (o) => o.meta.source === l);
  }), e;
}
function Ti(e, i, t = !0, n) {
  return tn(e, i, []), t && (n = n || `${e.name}-set`, e.walk((l) => {
    l.store.filter((s) => !(s.type === "error" && s.meta && s.meta.source === n));
  })), e;
}
function Di(e) {
  return j(e.props, "id") || (e.props.id = `input_${ui++}`), e;
}
function Bi(e) {
  const i = {
    initial: typeof e == "object" ? de(e) : e
  };
  let t, n = !0;
  return new Proxy(i, {
    get(...l) {
      const [s, r] = l;
      if (j(i, r))
        return Reflect.get(...l);
      if (t && typeof r == "string" && t.config[r] !== void 0)
        return t.config[r];
    },
    set(l, s, r, o) {
      if (s === "_n")
        return t = r, !0;
      if (s === "_emit")
        return n = r, !0;
      const { prop: u, value: d } = t.hook.prop.dispatch({
        prop: s,
        value: r
      });
      if (!ue(i[u], d, !1) || typeof d == "object") {
        const _ = Reflect.set(l, u, d, o);
        return n && (t.emit("prop", { prop: u, value: d }), typeof u == "string" && t.emit(`prop:${u}`, d)), _;
      }
      return !0;
    }
  });
}
function zi(e, i, t, n) {
  return i.traps.set(t, n), e;
}
function Ui(e, i) {
  if (e.props.definition)
    return e.define(e.props.definition);
  for (const t of i) {
    if (e.props.definition)
      return;
    typeof t.library == "function" && t.library(e);
  }
}
function Ki(e) {
  const i = Qt(e), t = en(e.config || {}, e.parent);
  return {
    _d: 0,
    _e: Kt(),
    uid: Symbol(),
    _resolve: !1,
    _tmo: !1,
    _value: i,
    children: In(e.children || []),
    config: t,
    hook: oi(),
    isCreated: !1,
    isSettled: !0,
    ledger: Zn(),
    name: fi(e),
    parent: e.parent || null,
    plugins: /* @__PURE__ */ new Set(),
    props: Bi(i),
    settled: Promise.resolve(i),
    store: Dn(!0),
    sync: e.sync || !1,
    traps: ri(),
    type: e.type || "input",
    value: i
  };
}
function qi(e, i) {
  var t;
  if (e.ledger.init(e.store._n = e.props._n = e.config._n = e), e.props._emit = !1, i.props && Object.assign(e.props, i.props), e.props._emit = !0, Ui(e, /* @__PURE__ */ new Set([
    ...i.plugins || [],
    ...e.parent ? e.parent.plugins : []
  ])), i.plugins)
    for (const n of i.plugins)
      _t(e, e._c, n, !0, !1);
  return Di(e), e.each((n) => e.add(n)), e.parent && e.parent.add(e, i.index), e.type === "input" && e.children.length && W(100, e), Xt(e, e._c, e._value, !1), e.store.release(), !((t = i.props) === null || t === void 0) && t.id && Xn(e), e.emit("created", e), e.isCreated = !0, e;
}
function Wi(e) {
  var i, t, n, l;
  return {
    __FKP: !0,
    uid: Symbol(),
    name: (i = e == null ? void 0 : e.name) !== null && i !== void 0 ? i : `p_${Yt++}`,
    value: (t = e == null ? void 0 : e.value) !== null && t !== void 0 ? t : null,
    _value: (n = e == null ? void 0 : e.value) !== null && n !== void 0 ? n : null,
    type: (l = e == null ? void 0 : e.type) !== null && l !== void 0 ? l : "input",
    use: () => {
    },
    input(s) {
      return this._value = s, this.value = s, Promise.resolve();
    },
    isSettled: !0
  };
}
function Ni(e) {
  return "__FKP" in e;
}
function Hi(e) {
  const i = e || {}, t = Ki(i), n = new Proxy(t, {
    get(...l) {
      const [, s] = l;
      if (s === "__FKNode__")
        return !0;
      const r = t.traps.get(s);
      return r && r.get ? r.get(n, t) : Reflect.get(...l);
    },
    set(...l) {
      const [, s, r] = l, o = t.traps.get(s);
      return o && o.set ? o.set(n, t, s, r) : Reflect.set(...l);
    }
  });
  return qi(n, i);
}
function ut(e) {
  return typeof e != "string" && j(e, "$el");
}
function ft(e) {
  return typeof e != "string" && j(e, "$cmp");
}
function be(e) {
  return !e || typeof e == "string" ? !1 : j(e, "if") && j(e, "then");
}
function Zi(e) {
  return typeof e != "string" && "$formkit" in e;
}
function Ji(e) {
  if (typeof e == "string")
    return {
      $el: "text",
      children: e
    };
  if (Zi(e)) {
    const { $formkit: i, for: t, if: n, children: l, bind: s, ...r } = e;
    return Object.assign({
      $cmp: "FormKit",
      props: { ...r, type: i }
    }, n ? { if: n } : {}, t ? { for: t } : {}, l ? { children: l } : {}, s ? { bind: s } : {});
  }
  return e;
}
function Z(e) {
  let i;
  const t = /* @__PURE__ */ new Set(), n = function(m, p) {
    return typeof m == "function" ? m(p) : m;
  }, l = [
    {
      "&&": (f, m, p) => n(f, p) && n(m, p),
      "||": (f, m, p) => n(f, p) || n(m, p)
    },
    {
      "===": (f, m, p) => n(f, p) === n(m, p),
      "!==": (f, m, p) => n(f, p) !== n(m, p),
      "==": (f, m, p) => n(f, p) == n(m, p),
      "!=": (f, m, p) => n(f, p) != n(m, p),
      ">=": (f, m, p) => n(f, p) >= n(m, p),
      "<=": (f, m, p) => n(f, p) <= n(m, p),
      ">": (f, m, p) => n(f, p) > n(m, p),
      "<": (f, m, p) => n(f, p) < n(m, p)
    },
    {
      "+": (f, m, p) => n(f, p) + n(m, p),
      "-": (f, m, p) => n(f, p) - n(m, p)
    },
    {
      "*": (f, m, p) => n(f, p) * n(m, p),
      "/": (f, m, p) => n(f, p) / n(m, p),
      "%": (f, m, p) => n(f, p) % n(m, p)
    }
  ], s = l.reduce((f, m) => f.concat(Object.keys(m)), []), r = new Set(s.map((f) => f.charAt(0)));
  function o(f, m, p, h) {
    const $ = f.filter((v) => v.startsWith(m));
    return $.length ? $.find((v) => h.length >= p + v.length && h.substring(p, p + v.length) === v ? v : !1) : !1;
  }
  function u(f, m, p = 1) {
    let h = p ? m.substring(f + 1).trim() : m.substring(0, f).trim();
    if (!h.length)
      return -1;
    if (!p) {
      const v = h.split("").reverse(), b = v.findIndex((c) => r.has(c));
      h = v.slice(b).join("");
    }
    const $ = h[0];
    return l.findIndex((v) => {
      const b = Object.keys(v);
      return !!o(b, $, 0, h);
    });
  }
  function d(f, m) {
    let p = "";
    const h = m.length;
    let $ = 0;
    for (let v = f; v < h; v++) {
      const b = m.charAt(v);
      if (b === "(")
        $++;
      else if (b === ")")
        $--;
      else if ($ === 0 && b === " ")
        continue;
      if ($ === 0 && o(s, b, v, m))
        return [p, v - 1];
      p += b;
    }
    return [p, m.length - 1];
  }
  function _(f, m = 0) {
    const p = l[m], h = f.length, $ = Object.keys(p);
    let v = 0, b = !1, c = null, g = "", k = null, L, R = "", C = "", F = "", B = "", H = 0;
    const N = (P, q) => {
      P ? F += q : g += q;
    };
    for (let P = 0; P < h; P++)
      if (R = C, C = f.charAt(P), (C === "'" || C === '"') && R !== "\\" && (v === 0 && !b || v && !B)) {
        v ? B = C : b = C, N(v, C);
        continue;
      } else if (b && (C !== b || R === "\\") || B && (C !== B || R === "\\")) {
        N(v, C);
        continue;
      } else if (b === C) {
        b = !1, N(v, C);
        continue;
      } else if (B === C) {
        B = !1, N(v, C);
        continue;
      } else {
        if (C === " ")
          continue;
        if (C === "(")
          v === 0 ? H = P : F += C, v++;
        else if (C === ")")
          if (v--, v === 0) {
            const q = typeof g == "string" && g.startsWith("$") ? g : void 0, ke = q && f.charAt(P + 1) === ".";
            let le = "";
            ke && ([le, P] = d(P + 2, f));
            const Re = c ? m : u(H, f, 0), Ce = u(P, f);
            Re === -1 && Ce === -1 ? g = V(F, -1, q, le) : c && (Re >= Ce || Ce === -1) && m === Re ? (k = c.bind(null, V(F, -1, q, le)), c = null, g = "") : Ce > Re && m === Ce ? g = V(F, -1, q, le) : g += `(${F})${ke ? `.${le}` : ""}`, F = "";
          } else
            F += C;
        else if (v === 0 && (L = o($, C, P, f))) {
          P === 0 && W(103, [L, f]), P += L.length - 1, P === f.length - 1 && W(104, [L, f]), c ? g && (k = c.bind(null, V(g, m)), c = p[L].bind(null, k), g = "") : k ? (c = p[L].bind(null, V(k, m)), k = null) : (c = p[L].bind(null, V(g, m)), g = "");
          continue;
        } else
          N(v, C);
      }
    return g && c && (c = c.bind(null, V(g, m))), c = !c && k ? k : c, !c && g && (c = (P, q) => typeof P == "function" ? P(q) : P, c = c.bind(null, V(g, m))), !c && !g && W(105, f), c;
  }
  function V(f, m, p, h) {
    if (p) {
      const $ = V(p, l.length);
      let v, b = h ? Z(`$${h}`) : !1;
      if (typeof $ == "function") {
        const c = On(String(f)).map((g) => V(g, -1));
        return (g) => {
          const k = $(g);
          return typeof k != "function" ? (je(150, p), k) : (v = k(...c.map((L) => typeof L == "function" ? L(g) : L)), b && (b = b.provide((L) => {
            const R = i(L);
            return L.reduce((F, B) => {
              if (B === h || (h == null ? void 0 : h.startsWith(`${B}(`))) {
                const N = xn(v, B);
                F[B] = () => N;
              } else
                F[B] = R[B];
              return F;
            }, {});
          })), b ? b() : v);
        };
      }
    } else if (typeof f == "string") {
      if (f === "true")
        return !0;
      if (f === "false")
        return !1;
      if (f === "undefined")
        return;
      if (Ln(f))
        return An(f.substring(1, f.length - 1));
      if (!isNaN(+f))
        return Number(f);
      if (m < l.length - 1)
        return _(f, m + 1);
      if (f.startsWith("$")) {
        const $ = f.substring(1);
        return t.add($), function(b) {
          return $ in b ? b[$]() : void 0;
        };
      }
      return f;
    }
    return f;
  }
  const J = _(e.startsWith("$:") ? e.substring(2) : e), y = Array.from(t);
  function I(f) {
    return i = f, Object.assign(J.bind(null, f(y)), {
      provide: I
    });
  }
  return Object.assign(J, {
    provide: I
  });
}
const Ot = "0.17.3";
function Yi(e) {
  let i = 1;
  return Array.isArray(e) ? e.map((t) => typeof t == "string" || typeof t == "number" ? {
    label: String(t),
    value: String(t)
  } : (typeof t == "object" && "value" in t && typeof t.value != "string" && Object.assign(t, {
    value: `__mask_${i++}`,
    __original: t.value
  }), t)) : Object.keys(e).map((t) => ({
    label: e[t],
    value: t
  }));
}
function ge(e, i) {
  if (Array.isArray(e)) {
    for (const t of e)
      if (i == t.value)
        return "__original" in t ? t.__original : t.value;
  }
  return i;
}
function $e(e, i) {
  return e === null && i === void 0 || e === void 0 && i === null ? !1 : e == i ? !0 : _e(e) && _e(i) ? ue(e, i) : !1;
}
function gt(e) {
  e.hook.prop((i, t) => (i.prop === "options" && (typeof i.value == "function" ? (e.props.optionsLoader = i.value, i.value = []) : i.value = Yi(i.value)), t(i)));
}
function w(e, i, t = !1) {
  return (...n) => {
    const l = (s) => {
      const r = !i || typeof i == "string" ? { $el: i } : i();
      return (ut(r) || ft(r)) && (r.meta || (r.meta = { section: e }), n.length && !r.children && (r.children = [
        ...n.map((o) => typeof o == "string" ? o : o(s))
      ]), ut(r) && (r.attrs = {
        class: `$classes.${e}`,
        ...r.attrs || {}
      })), {
        if: `$slots.${e}`,
        then: `$slots.${e}`,
        else: e in s ? Se(r, s[e]) : r
      };
    };
    return l._s = e, t ? Qi(l) : l;
  };
}
function Qi(e) {
  return (i) => [e(i)];
}
function Ke(e) {
  return typeof e == "object" && ("$el" in e || "$cmp" in e || "$formkit" in e);
}
function Se(e, i = {}) {
  return typeof e == "string" ? Ke(i) || typeof i == "string" ? i : e : Array.isArray(e) ? Ke(i) ? i : e : Ye(e, i);
}
const fe = w("outer", () => ({
  $el: "div",
  attrs: {
    key: "$id",
    "data-family": "$family || undefined",
    "data-type": "$type",
    "data-multiple": '$attrs.multiple || ($type != "select" && $options != undefined) || undefined',
    "data-disabled": "$disabled || undefined",
    "data-complete": "$state.complete || undefined",
    "data-invalid": "$state.valid === false && $state.validationVisible || undefined",
    "data-errors": "$state.errors || undefined",
    "data-submitted": "$state.submitted || undefined",
    "data-prefix-icon": "$_rawPrefixIcon !== undefined || undefined",
    "data-suffix-icon": "$_rawSuffixIcon !== undefined || undefined",
    "data-prefix-icon-click": "$onPrefixIconClick !== undefined || undefined",
    "data-suffix-icon-click": "$onSuffixIconClick !== undefined || undefined"
  }
})), te = w("inner", "div"), we = w("wrapper", "div"), Ee = w("label", () => ({
  $el: "label",
  if: "$label",
  attrs: {
    for: "$id"
  }
})), ne = w("messages", () => ({
  $el: "ul",
  if: "$defaultMessagePlacement && $fns.length($messages)"
})), ie = w("message", () => ({
  $el: "li",
  for: ["message", "$messages"],
  attrs: {
    key: "$message.key",
    id: "$id + '-' + $message.key",
    "data-message-type": "$message.type"
  }
})), X = w("prefix", null), G = w("suffix", null), ee = w("help", () => ({
  $el: "div",
  if: "$help",
  attrs: {
    id: '$: "help-" + $id'
  }
})), nn = w("fieldset", () => ({
  $el: "fieldset",
  attrs: {
    id: "$id",
    "aria-describedby": {
      if: "$help",
      then: '$: "help-" + $id',
      else: void 0
    }
  }
})), qe = w("decorator", () => ({
  $el: "span",
  attrs: {
    "aria-hidden": "true"
  }
})), We = w("input", () => ({
  $el: "input",
  bind: "$attrs",
  attrs: {
    type: "$type",
    name: "$node.props.altName || $node.name",
    disabled: "$option.attrs.disabled || $disabled",
    onInput: "$handlers.toggleChecked",
    checked: "$fns.eq($_value, $onValue)",
    onBlur: "$handlers.blur",
    value: "$: true",
    id: "$id",
    "aria-describedby": {
      if: "$options.length",
      then: {
        if: "$option.help",
        then: '$: "help-" + $option.attrs.id',
        else: void 0
      },
      else: {
        if: "$help",
        then: '$: "help-" + $id',
        else: void 0
      }
    }
  }
})), ln = w("legend", () => ({
  $el: "legend",
  if: "$label"
})), an = w("option", () => ({
  $el: "li",
  for: ["option", "$options"],
  attrs: {
    "data-disabled": "$option.attrs.disabled || $disabled"
  }
})), sn = w("options", "ul"), Ne = w("wrapper", () => ({
  $el: "label",
  attrs: {
    "data-disabled": {
      if: "$options.length",
      then: void 0,
      else: "$disabled || undefined"
    },
    "data-checked": {
      if: "$options == undefined",
      then: "$fns.eq($_value, $onValue) || undefined",
      else: "$fns.isChecked($option.value) || undefined"
    }
  }
})), rn = w("optionHelp", () => ({
  $el: "div",
  if: "$option.help",
  attrs: {
    id: '$: "help-" + $option.attrs.id'
  }
})), He = w("label", "span"), Xi = w("input", () => ({
  $el: "button",
  bind: "$attrs",
  attrs: {
    type: "$type",
    disabled: "$disabled",
    name: "$node.name",
    id: "$id"
  }
})), Gi = w("default", null), el = w("input", () => ({
  $el: "input",
  bind: "$attrs",
  attrs: {
    type: "file",
    disabled: "$disabled",
    name: "$node.name",
    onChange: "$handlers.files",
    onBlur: "$handlers.blur",
    id: "$id",
    "aria-describedby": "$describedBy"
  }
})), tl = w("fileItem", () => ({
  $el: "li",
  for: ["file", "$value"]
})), nl = w("fileList", () => ({
  $el: "ul",
  if: "$value.length",
  attrs: {
    "data-has-multiple": {
      if: "$value.length > 1",
      then: "true"
    }
  }
})), il = w("fileName", () => ({
  $el: "span",
  attrs: {
    class: "$classes.fileName"
  }
})), Pt = w("fileRemove", () => ({
  $el: "button",
  attrs: {
    onClick: "$handlers.resetFiles"
  }
})), ll = w("noFiles", () => ({
  $el: "span",
  if: "$value.length == 0"
})), al = w("form", () => ({
  $el: "form",
  bind: "$attrs",
  attrs: {
    id: "$id",
    name: "$node.name",
    onSubmit: "$handlers.submit",
    "data-loading": "$state.loading || undefined"
  }
})), sl = w("actions", () => ({
  $el: "div",
  if: "$actions"
})), rl = w("submit", () => ({
  $cmp: "FormKit",
  bind: "$submitAttrs",
  props: {
    type: "submit",
    disabled: "$disabled",
    label: "$submitLabel"
  }
})), on = w("input", () => ({
  $el: "input",
  bind: "$attrs",
  attrs: {
    type: "$type",
    disabled: "$disabled",
    name: "$node.name",
    onInput: "$handlers.DOMInput",
    onBlur: "$handlers.blur",
    value: "$_value",
    id: "$id",
    "aria-describedby": "$describedBy"
  }
})), un = w("wrapper", null, !0), ol = w("input", () => ({
  $el: "select",
  bind: "$attrs",
  attrs: {
    id: "$id",
    "data-placeholder": "$fns.showPlaceholder($_value, $placeholder)",
    disabled: "$disabled",
    class: "$classes.input",
    name: "$node.name",
    onChange: "$handlers.onChange",
    onInput: "$handlers.selectInput",
    onBlur: "$handlers.blur",
    "aria-describedby": "$describedBy"
  }
})), ul = w("option", () => ({
  $el: "option",
  for: ["option", "$options"],
  bind: "$option.attrs",
  attrs: {
    class: "$classes.option",
    value: "$option.value",
    selected: "$fns.isSelected($option)"
  }
})), fl = () => ({
  $el: null,
  if: "$options.length",
  for: ["option", "$options"],
  children: "$slots.option"
}), cl = w("input", () => ({
  $el: "textarea",
  bind: "$attrs",
  attrs: {
    disabled: "$disabled",
    name: "$node.name",
    onInput: "$handlers.DOMInput",
    onBlur: "$handlers.blur",
    value: "$_value",
    id: "$id",
    "aria-describedby": "$describedBy"
  },
  children: "$initialValue"
})), z = (e, i) => w(`${e}Icon`, () => {
  const t = `_raw${e.charAt(0).toUpperCase()}${e.slice(1)}Icon`;
  return {
    if: `$${e}Icon && $${t}`,
    $el: `${i || "span"}`,
    attrs: {
      class: `$classes.${e}Icon + " formkit-icon"`,
      innerHTML: `$${t}`,
      onClick: `$handlers.iconClick(${e})`,
      for: {
        if: `${i === "label"}`,
        then: "$id"
      }
    }
  };
})();
function fn(e) {
  return function(i, t) {
    return i.prop === "options" && Array.isArray(i.value) && (i.value = i.value.map((n) => {
      var l;
      return !((l = n.attrs) === null || l === void 0) && l.id ? n : Ye(n, {
        attrs: {
          id: `${e.name}-option-${pt(String(n.value))}`
        }
      });
    }), e.props.type === "checkbox" && !Array.isArray(e.value) && (e.isCreated ? e.input([], !1) : e.on("created", () => {
      Array.isArray(e.value) || e.input([], !1);
    }))), t(i);
  };
}
function dl(e, i) {
  const t = i.target;
  if (t instanceof HTMLInputElement) {
    const n = Array.isArray(e.props.options) ? ge(e.props.options, t.value) : t.value;
    Array.isArray(e.props.options) && e.props.options.length ? Array.isArray(e._value) ? e._value.some((l) => $e(n, l)) ? e.input(e._value.filter((l) => !$e(n, l))) : e.input([...e._value, n]) : e.input([n]) : t.checked ? e.input(e.props.onValue) : e.input(e.props.offValue);
  }
}
function pl(e, i) {
  var t, n;
  return (t = e.context) === null || t === void 0 || t.value, (n = e.context) === null || n === void 0 || n._value, Array.isArray(e._value) ? e._value.some((l) => $e(ge(e.props.options, i), l)) : !1;
}
function ml(e) {
  e.on("created", () => {
    var i, t;
    !((i = e.context) === null || i === void 0) && i.handlers && (e.context.handlers.toggleChecked = dl.bind(null, e)), !((t = e.context) === null || t === void 0) && t.fns && (e.context.fns.isChecked = pl.bind(null, e)), j(e.props, "onValue") || (e.props.onValue = !0), j(e.props, "offValue") || (e.props.offValue = !1);
  }), e.hook.prop(fn(e));
}
function Xe(e) {
  e.on("created", () => {
    e.props.disabled = Y(e.props.disabled), e.config.disabled = Y(e.props.disabled);
  }), e.hook.prop(({ prop: i, value: t }, n) => (t = i === "disabled" ? Y(t) : t, n({ prop: i, value: t }))), e.on("prop:disabled", ({ payload: i }) => {
    e.config.disabled = Y(i);
  });
}
function Be(e, i) {
  return (t) => {
    t.store.set(re({
      key: e,
      type: "ui",
      value: i || e,
      meta: {
        localize: !0,
        i18nArgs: [t]
      }
    }));
  };
}
const et = typeof window < "u";
function cn(e) {
  e.target instanceof HTMLElement && e.target.hasAttribute("data-file-hover") && e.target.removeAttribute("data-file-hover");
}
function xt(e, i) {
  i.target instanceof HTMLInputElement ? e === "dragover" && i.target.setAttribute("data-file-hover", "true") : i.preventDefault(), e === "drop" && cn(i);
}
function bl(e) {
  Be("noFiles", "Select file")(e), Be("removeAll", "Remove all")(e), Be("remove")(e), et && (window._FormKit_File_Drop || (window.addEventListener("dragover", xt.bind(null, "dragover")), window.addEventListener("drop", xt.bind(null, "drop")), window.addEventListener("dragleave", cn), window._FormKit_File_Drop = !0)), e.hook.input((i, t) => t(Array.isArray(i) ? i : [])), e.on("reset", () => {
    if (e.props.id && et) {
      const i = document.getElementById(e.props.id);
      i && (i.value = "");
    }
  }), e.on("created", () => {
    Array.isArray(e.value) || e.input([], !1), e.context && (e.context.handlers.resetFiles = (i) => {
      if (i.preventDefault(), e.input([]), e.props.id && et) {
        const t = document.getElementById(e.props.id);
        t && (t.value = "");
      }
    }, e.context.handlers.files = (i) => {
      var t, n;
      const l = [];
      if (i.target instanceof HTMLInputElement && i.target.files) {
        for (let s = 0; s < i.target.files.length; s++) {
          let r;
          (r = i.target.files.item(s)) && l.push({ name: r.name, file: r });
        }
        e.input(l);
      }
      e.context && (e.context.files = l), typeof ((t = e.props.attrs) === null || t === void 0 ? void 0 : t.onChange) == "function" && ((n = e.props.attrs) === null || n === void 0 || n.onChange(i));
    });
  });
}
async function yl(e, i) {
  i.preventDefault(), await e.settled;
  const t = (n) => n.store.set(re({
    key: "submitted",
    value: !0,
    visible: !1
  }));
  if (e.walk(t), t(e), typeof e.props.onSubmitRaw == "function" && e.props.onSubmitRaw(i, e), e.ledger.value("blocking"))
    typeof e.props.onSubmitInvalid == "function" && e.props.onSubmitInvalid(e), e.props.incompleteMessage !== !1 && e.store.set(re({
      blocking: !1,
      key: "incomplete",
      meta: {
        localize: e.props.incompleteMessage === void 0,
        i18nArgs: [{ node: e }],
        showAsMessage: !0
      },
      type: "ui",
      value: e.props.incompleteMessage || "Form incomplete."
    }));
  else if (typeof e.props.onSubmit == "function") {
    const n = e.props.onSubmit(e.hook.submit.dispatch(Pe(e.value)), e);
    if (n instanceof Promise) {
      const l = e.props.disabled === void 0 && e.props.submitBehavior !== "live";
      l && (e.props.disabled = !0), e.store.set(re({
        key: "loading",
        value: !0,
        visible: !1
      })), await n, l && (e.props.disabled = !1), e.store.remove("loading");
    }
  } else
    i.target instanceof HTMLFormElement && i.target.submit();
}
function vl(e) {
  e.props.isForm = !0, e.on("created", () => {
    var i;
    !((i = e.context) === null || i === void 0) && i.handlers && (e.context.handlers.submit = yl.bind(null, e)), j(e.props, "actions") || (e.props.actions = !0);
  }), e.on("settled:blocking", () => e.store.remove("incomplete"));
}
function hl(e) {
  e.props.ignore === void 0 && (e.props.ignore = !0, e.parent = null);
}
function _l(e) {
  e.on("created", () => {
    e.context && (e.context.initialValue = e.value || "");
  });
}
function gl(e, i) {
  i.target instanceof HTMLInputElement && e.input(ge(e.props.options, i.target.value));
}
function $l(e, i) {
  var t, n;
  return (t = e.context) === null || t === void 0 || t.value, (n = e.context) === null || n === void 0 || n._value, $e(ge(e.props.options, i), e._value);
}
function wl(e) {
  e.on("created", () => {
    var i, t;
    Array.isArray(e.props.options) || je(350, {
      node: e,
      inputType: "radio"
    }), !((i = e.context) === null || i === void 0) && i.handlers && (e.context.handlers.toggleChecked = gl.bind(null, e)), !((t = e.context) === null || t === void 0) && t.fns && (e.context.fns.isChecked = $l.bind(null, e));
  }), e.hook.prop(fn(e));
}
function kl(e, i) {
  e.context && e.context.value;
  const t = "__original" in i ? i.__original : i.value;
  function n() {
    return !e.props.options.some((l) => ("__original" in l ? l.__original : l.value) === null);
  }
  return Array.isArray(e._value) ? e._value.some((l) => $e(l, t)) : (e._value === void 0 || e._value === null && n()) && i.attrs && i.attrs["data-is-placeholder"] ? !0 : $e(t, e._value);
}
async function Cl(e, i) {
  var t;
  typeof ((t = e.props.attrs) === null || t === void 0 ? void 0 : t.onChange) == "function" && (await new Promise((n) => setTimeout(n, 0)), await e.settled, e.props.attrs.onChange(i));
}
function Sl(e, i) {
  const t = i.target, n = t.hasAttribute("multiple") ? Array.from(t.selectedOptions).map((l) => ge(e.props.options, l.value)) : ge(e.props.options, t.value);
  e.input(n);
}
function Mt(e, i) {
  return e.some((t) => t.attrs && t.attrs["data-is-placeholder"]) ? e : [
    {
      label: i,
      value: "",
      attrs: {
        hidden: !0,
        disabled: !0,
        "data-is-placeholder": "true"
      }
    },
    ...e
  ];
}
function Vl(e) {
  e.on("created", () => {
    var i, t, n;
    const l = Y((i = e.props.attrs) === null || i === void 0 ? void 0 : i.multiple);
    !l && e.props.placeholder && Array.isArray(e.props.options) && (e.hook.prop(({ prop: s, value: r }, o) => (s === "options" && (r = Mt(r, e.props.placeholder)), o({ prop: s, value: r }))), e.props.options = Mt(e.props.options, e.props.placeholder)), l ? e.value === void 0 && e.input([], !1) : e.context && !e.context.options && (e.props.attrs = Object.assign({}, e.props.attrs, {
      value: e._value
    }), e.on("input", ({ payload: s }) => {
      e.props.attrs = Object.assign({}, e.props.attrs, {
        value: s
      });
    })), !((t = e.context) === null || t === void 0) && t.handlers && (e.context.handlers.selectInput = Sl.bind(null, e), e.context.handlers.onChange = Cl.bind(null, e)), !((n = e.context) === null || n === void 0) && n.fns && (e.context.fns.isSelected = kl.bind(null, e), e.context.fns.showPlaceholder = (s, r) => {
      if (!Array.isArray(e.props.options))
        return !1;
      const o = e.props.options.some((u) => {
        if (u.attrs && "data-is-placeholder" in u.attrs)
          return !1;
        const d = "__original" in u ? u.__original : u.value;
        return ue(s, d);
      });
      return r && !o ? !0 : void 0;
    });
  }), e.hook.input((i, t) => {
    var n, l, s;
    return !e.props.placeholder && i === void 0 && Array.isArray((n = e.props) === null || n === void 0 ? void 0 : n.options) && e.props.options.length && !Y((s = (l = e.props) === null || l === void 0 ? void 0 : l.attrs) === null || s === void 0 ? void 0 : s.multiple) && (i = "__original" in e.props.options[0] ? e.props.options[0].__original : e.props.options[0].value), t(i);
  });
}
function ve(e, i) {
  return (t) => {
    t.props[`${e}Icon`] === void 0 && (t.props[`${e}Icon`] = `default:${i}`);
  };
}
function ct(e) {
  return !!(be(e) && e.if && e.if.startsWith("$slots.") && typeof e.then == "string" && e.then.startsWith("$slots.") && "else" in e);
}
function Il(e) {
  return fe(we(Ee("$label"), te(X(), e(), G())), ee("$help"), ne(ie("$message.value")));
}
function Q(e, i, t) {
  const n = (l) => {
    const s = i(l);
    if (t || Ke(s) && "if" in s || ct(s)) {
      const r = {
        if: e,
        then: s
      };
      return t && (r.else = t(l)), r;
    } else
      ct(s) ? Object.assign(s.else, { if: e }) : Ke(s) && Object.assign(s, { if: e });
    return s;
  };
  return n._s = Je(), n;
}
function Ie(e, i) {
  const t = (n) => {
    const l = e({});
    return ct(l) ? (Array.isArray(l.else) || (l.else = Se(Se(l.else, i), e._s ? n[e._s] : {})), l) : Se(Se(l, i), e._s ? n[e._s] : {});
  };
  return t._s = e._s, t;
}
fe(ne(ie("$message.value")), we(Xi(z("prefix"), X(), Gi("$label || $ui.submit.value"), G(), z("suffix"))), ee("$help")), Be("submit");
fe(
  Q(
    "$options == undefined",
    /**
     * Single checkbox structure.
     */
    Ne(te(X(), We(), qe(z("decorator")), G()), Ie(He("$label"), {
      if: "$label"
    })),
    /**
     * Multi checkbox structure.
     */
    nn(ln("$label"), ee("$help"), sn(an(Ne(te(X(), Ie(We(), {
      bind: "$option.attrs",
      attrs: {
        id: "$option.attrs.id",
        value: "$option.value",
        checked: "$fns.isChecked($option.value)"
      }
    }), qe(z("decorator")), G()), Ie(He("$option.label"), {
      if: "$option.label"
    })), rn("$option.help"))))
  ),
  // Help text only goes under the input when it is a single.
  Q("$options == undefined && $help", ee("$help")),
  ne(ie("$message.value"))
), ve("decorator", "checkboxDecorator");
fe(we(Ee("$label"), te(z("prefix", "label"), X(), el(), nl(tl(z("fileItem"), il("$file.name"), Q("$value.length === 1", Pt(z("fileRemove"), "$ui.remove.value")))), Q("$value.length > 1", Pt("$ui.removeAll.value")), ll(z("noFiles"), "$ui.noFiles.value"), G(), z("suffix"))), ee("$help"), ne(ie("$message.value"))), ve("fileItem", "fileItem"), ve("fileRemove", "fileRemove"), ve("noFiles", "noFiles");
al("$slots.default", ne(ie("$message.value")), sl(rl()));
un("$slots.default");
on();
un("$slots.default");
fe(
  Q(
    "$options == undefined",
    /**
     * Single radio structure.
     */
    Ne(te(X(), We(), qe(z("decorator")), G()), Q("$label", He("$label"))),
    /**
     * Multi radio structure.
     */
    nn(ln("$label"), ee("$help"), sn(an(Ne(te(X(), Ie(We(), {
      bind: "$option.attrs",
      attrs: {
        id: "$option.attrs.id",
        value: "$option.value",
        checked: "$fns.isChecked($option.value)"
      }
    }), qe(z("decorator")), G()), Ie(He("$option.label"), {
      if: "$option.label"
    })), rn("$option.help"))))
  ),
  // Help text only goes under the input when it is a single.
  Q("$options == undefined && $help", ee("$help")),
  ne(ie("$message.value"))
), ve("decorator", "radioDecorator");
fe(we(Ee("$label"), te(z("prefix"), X(), ol(Q("$slots.default", () => "$slots.default", Q("$slots.option", fl, ul("$option.label")))), Q("$attrs.multiple !== undefined", () => "", z("select")), G(), z("suffix"))), ee("$help"), ne(ie("$message.value"))), ve("select", "select");
fe(we(Ee("$label"), te(z("prefix", "label"), X(), cl(), G(), z("suffix"))), ee("$help"), ne(ie("$message.value")));
fe(we(Ee("$label"), te(z("prefix", "label"), X(), on(), G(), z("suffix"))), ee("$help"), ne(ie("$message.value")));
re({
  type: "state",
  blocking: !0,
  visible: !1,
  value: !0,
  key: "validating"
});
let ze;
const Ll = new Promise((e) => {
}), Ge = typeof window < "u" && typeof fetch < "u";
ze = Ge ? getComputedStyle(document.documentElement) : void 0;
const Ve = {}, tt = {};
function nt(e, i) {
  return (t) => {
    if (typeof t == "boolean")
      return;
    if (t.startsWith("<svg"))
      return t;
    if (typeof t != "string")
      return;
    const n = t.startsWith("default:");
    t = n ? t.split(":")[1] : t;
    const l = t in Ve;
    let s;
    if (l)
      return Ve[t];
    if (!tt[t]) {
      if (s = Al(t), s = Ge && typeof s > "u" ? Promise.resolve(s) : s, s instanceof Promise)
        tt[t] = s.then((r) => !r && typeof t == "string" && !n ? s = typeof e == "function" ? e(t) : Pl(t, i) : r).then((r) => (typeof t == "string" && (Ve[n ? `default:${t}` : t] = r), r));
      else if (typeof s == "string")
        return Ve[n ? `default:${t}` : t] = s, s;
    }
    return tt[t];
  };
}
function Al(e) {
  if (Ge)
    return Ll.then(() => Ol(e));
}
function Ol(e) {
  const i = ze == null ? void 0 : ze.getPropertyValue(`--fk-icon-${e}`);
  if (i) {
    const t = atob(i);
    if (t.startsWith("<svg"))
      return Ve[e] = t, t;
  }
}
function Pl(e, i) {
  const t = Ot.startsWith("__") ? "latest" : Ot, n = typeof i == "function" ? i(e) : `https://cdn.jsdelivr.net/npm/@formkit/icons@${t}/dist/icons/${e}.svg`;
  if (Ge)
    return fetch(`${n}`).then(async (l) => {
      const s = await l.text();
      if (s.startsWith("<svg"))
        return s;
    }).catch((l) => {
      console.error(l);
    });
}
let $t = !1;
const jt = {
  /**
   * FormKit errors:
   */
  100: ({ data: e }) => `Only groups, lists, and forms can have children (${e.name}).`,
  101: ({ data: e }) => `You cannot directly modify the store (${e.name}). See: https://formkit.com/advanced/core#message-store`,
  102: ({ data: [e, i] }) => `You cannot directly assign node.${i} (${e.name})`,
  103: ({ data: [e] }) => `Schema expressions cannot start with an operator (${e})`,
  104: ({ data: [e, i] }) => `Schema expressions cannot end with an operator (${e} in "${i}")`,
  105: ({ data: e }) => `Invalid schema expression: ${e}`,
  106: ({ data: e }) => `Cannot submit because (${e}) is not in a form.`,
  107: ({ data: [e, i] }) => `Cannot set ${e.name} to non object value: ${i}`,
  108: ({ data: [e, i] }) => `Cannot set ${e.name} to non array value: ${i}`,
  /**
   * Input specific errors:
   */
  300: ({ data: [e] }) => `Cannot set behavior prop to overscroll (on ${e.name} input) when options prop is a function.`,
  /**
   * FormKit vue errors:
   */
  600: ({ data: e }) => `Unknown input type${typeof e.props.type == "string" ? ' "' + e.props.type + '"' : ""} ("${e.name}")`,
  601: ({ data: e }) => `Input definition${typeof e.props.type == "string" ? ' "' + e.props.type + '"' : ""} is missing a schema or component property (${e.name}).`
}, Et = {
  /**
   * Core warnings:
   */
  150: ({ data: e }) => `Schema function "${e}()" is not a valid function.`,
  151: ({ data: e }) => `No form element with id: ${e}`,
  152: ({ data: e }) => `No input element with id: ${e}`,
  /**
   * Input specific warnings:
   */
  350: ({ data: { node: e, inputType: i } }) => `Invalid options prop for ${e.name} input (${i}). See https://formkit.com/inputs/${i}`,
  /**
   * Vue warnings:
   */
  650: 'Schema "$get()" must use the id of an input to access.',
  651: ({ data: e }) => `Cannot setErrors() on "${e}" because no such id exists.`,
  652: ({ data: e }) => `Cannot clearErrors() on "${e}" because no such id exists.`,
  /**
   * Deprecation warnings:
   */
  800: ({ data: e }) => `${e} is deprecated.`
}, xl = (e, i) => {
  if (e.code in jt) {
    const t = jt[e.code];
    e.message = typeof t == "function" ? t(e) : t;
  }
  return i(e);
};
$t || bt(xl);
const Ml = (e, i) => {
  if (e.code in Et) {
    const t = Et[e.code];
    e.message = typeof t == "function" ? t(e) : t;
  }
  return i(e);
};
$t || yt(Ml);
$t = !0;
const jl = typeof window > "u", it = /* @__PURE__ */ new Map();
function El(e, i) {
  var t;
  !jl || !e || (it.has(e) || it.set(e, /* @__PURE__ */ new Set()), (t = it.get(e)) === null || t === void 0 || t.add(i));
}
const dn = typeof window > "u", Le = {}, he = {};
let K;
const se = /* @__PURE__ */ new WeakMap(), Rl = "__raw__", Fl = /[a-zA-Z0-9\-][cC]lass$/;
function Tl(e, i) {
  const t = Me(null);
  if (e === "get") {
    const l = {};
    return t.value = Dl.bind(null, l), t;
  }
  const n = e.split(".");
  return ae(() => {
    t.value = wt(kn(i) ? i.value : i, n);
  }), t;
}
function wt(e, i) {
  if (Array.isArray(e)) {
    for (const l of e) {
      const s = l !== !1 && wt(l, i);
      if (s !== void 0)
        return s;
    }
    return;
  }
  let t, n = e;
  for (const l in i) {
    const s = i[l];
    if (typeof n != "object" || n === null) {
      t = void 0;
      break;
    }
    const r = n[s];
    if (Number(l) === i.length - 1 && r !== void 0) {
      t = typeof r == "function" ? r.bind(n) : r;
      break;
    }
    n = r;
  }
  return t;
}
function Dl(e, i) {
  if (typeof i != "string")
    return je(650);
  if (i in e || (e[i] = Me(void 0)), e[i].value === void 0) {
    e[i].value = null;
    const t = Ht(i);
    t && (e[i].value = t.context), ei(i, ({ payload: n }) => {
      e[i].value = Qe(n) ? n.context : n;
    });
  }
  return e[i].value;
}
function Rt(e, i, t) {
  function n(y, I) {
    const f = V(Z(I.if), { if: !0 }), m = d(y, I.then), p = I.else ? d(y, I.else) : null;
    return [f, m, p];
  }
  function l(y, I) {
    var f, m;
    const p = V(Z(y.if));
    let h = () => I, $ = () => I;
    return typeof y.then == "object" ? $ = s(y.then, void 0) : typeof y.then == "string" && (!((f = y.then) === null || f === void 0) && f.startsWith("$")) ? $ = V(Z(y.then)) : $ = () => y.then, j(y, "else") && (typeof y.else == "object" ? h = s(y.else) : typeof y.else == "string" && (!((m = y.else) === null || m === void 0) && m.startsWith("$")) ? h = V(Z(y.else)) : h = () => y.else), () => p() ? $() : h();
  }
  function s(y, I, f = {}) {
    const m = new Set(Object.keys(y || {})), p = I ? V(Z(I)) : () => ({}), h = [
      ($) => {
        const v = p();
        for (const b in v)
          m.has(b) || ($[b] = v[b]);
      }
    ];
    if (y) {
      if (be(y))
        return l(y, f);
      for (let $ in y) {
        const v = y[$];
        let b;
        const c = typeof v == "string";
        $.startsWith(Rl) ? ($ = $.substring(7), b = () => v) : c && v.startsWith("$") && v.length > 1 && !(v.startsWith("$reset") && Fl.test($)) ? b = V(Z(v)) : typeof v == "object" && be(v) ? b = l(v, void 0) : typeof v == "object" && _e(v) ? b = s(v) : b = () => v, h.push((g) => {
          g[$] = b();
        });
      }
    }
    return () => {
      const $ = Array.isArray(y) ? [] : {};
      return h.forEach((v) => v($)), $;
    };
  }
  function r(y, I) {
    let f = null, m = () => null, p = !1, h = null, $ = null, v = null, b = !1;
    const c = Ji(I);
    if (ut(c) ? (f = c.$el, m = c.$el !== "text" ? s(c.attrs, c.bind) : () => null) : ft(c) ? (typeof c.$cmp == "string" ? j(y, c.$cmp) ? f = y[c.$cmp] : (f = c.$cmp, b = !0) : f = c.$cmp, m = s(c.props, c.bind)) : be(c) && ([p, h, $] = n(y, c)), !be(c) && "if" in c ? p = V(Z(c.if)) : !be(c) && f === null && (p = () => !0), "children" in c && c.children)
      if (typeof c.children == "string")
        if (c.children.startsWith("$slots."))
          f = f === "text" ? "slot" : f, h = V(Z(c.children));
        else if (c.children.startsWith("$") && c.children.length > 1) {
          const g = V(Z(c.children));
          h = () => String(g());
        } else
          h = () => String(c.children);
      else if (Array.isArray(c.children))
        h = d(y, c.children);
      else {
        const [g, k, L] = n(y, c.children);
        h = (R) => g && g() ? k && k(R) : L && L(R);
      }
    if (ft(c))
      if (h) {
        const g = h;
        h = (k) => ({
          default(L, R) {
            var C, F, B, H;
            const N = K;
            R && (K = R), L && ((C = se.get(K)) === null || C === void 0 || C.unshift(L)), k && ((F = se.get(K)) === null || F === void 0 || F.unshift(k));
            const P = g(k);
            return L && ((B = se.get(K)) === null || B === void 0 || B.shift()), k && ((H = se.get(K)) === null || H === void 0 || H.shift()), K = N, P;
          }
        }), h.slot = !0;
      } else
        h = () => ({});
    if ("for" in c && c.for) {
      const g = c.for.length === 3 ? c.for[2] : c.for[1];
      v = [
        typeof g == "string" && g.startsWith("$") ? V(Z(g)) : () => g,
        c.for[0],
        c.for.length === 3 ? String(c.for[1]) : null
      ];
    }
    return [p, f, m, h, $, v, b];
  }
  function o(y, I) {
    const f = y(I), m = K;
    return Object.keys(f).reduce((p, h) => {
      const $ = f && f[h];
      return p[h] = (v) => $ && $(v, m) || null, p;
    }, {});
  }
  function u(y, I) {
    const [f, m, p, h, $, v, b] = r(y, I);
    let c = (g) => {
      if (f && m === null && h)
        return f() ? h(g) : $ && $(g);
      if (m && (!f || f())) {
        if (m === "text" && h)
          return Cn(String(h()));
        if (m === "slot" && h)
          return h(g);
        const k = b ? E(m) : m, L = h != null && h.slot ? o(h, g) : null;
        return Ae(k, p(), L || (h ? h(g) : []));
      }
      return typeof $ == "function" ? $(g) : $;
    };
    if (v) {
      const g = c, [k, L, R] = v;
      c = () => {
        const C = k(), F = Number.isFinite(C) ? Array(Number(C)).fill(0).map((P, q) => q) : C, B = [];
        if (typeof F != "object")
          return null;
        const H = se.get(K) || [], N = Array.isArray(F);
        for (const P in F) {
          if (N && P in Array.prototype)
            continue;
          const q = Object.defineProperty({
            ...H.reduce((ke, le) => ke.__idata ? { ...ke, ...le } : le, {}),
            [L]: F[P],
            ...R !== null ? { [R]: N ? Number(P) : P } : {}
          }, "__idata", { enumerable: !1, value: !0 });
          H.unshift(q), B.push(g.bind(null, q)()), H.shift();
        }
        return B;
      };
    }
    return c;
  }
  function d(y, I) {
    if (Array.isArray(I)) {
      const m = I.map(u.bind(null, y));
      return (p) => m.map((h) => h(p));
    }
    const f = u(y, I);
    return (m) => f(m);
  }
  const _ = [];
  function V(y, I = {}) {
    const f = /* @__PURE__ */ new WeakMap();
    return _.push((m, p) => {
      f.set(p, y.provide((h) => m(h, I)));
    }), () => f.get(K)();
  }
  function J(y, I) {
    var f;
    t ?? (t = JSON.stringify(i));
    const [m, p] = j(Le, t) ? Le[t] : [d(e, i), _];
    return dn || ((f = he[t]) !== null && f !== void 0 || (he[t] = 0), he[t]++, Le[t] = [m, p]), p.forEach((h) => {
      h(y, I);
    }), () => (K = I, m());
  }
  return J;
}
function pn(e, i) {
  const t = se.get(K) || [];
  let n;
  return t.length && (n = wt(t, e.split("."))), n === void 0 ? i : n;
}
function Bl(e, i) {
  return new Proxy(e, {
    get(...t) {
      let n;
      const l = t[1];
      if (typeof l == "string") {
        const s = K;
        K = i, n = pn(l, void 0), K = s;
      }
      return n !== void 0 ? n : Reflect.get(...t);
    }
  });
}
function Ft(e, i, t) {
  return e((n, l = {}) => n.reduce((s, r) => {
    if (r.startsWith("slots.")) {
      const o = r.substring(6), u = () => i.slots && j(i.slots, o) && typeof i.slots[o] == "function";
      if (l.if)
        s[r] = u;
      else if (i.slots) {
        const d = Bl(i, t);
        s[r] = () => u() ? i.slots[o](d) : null;
      }
    } else {
      const o = Tl(r, i);
      s[r] = () => pn(r, o.value);
    }
    return s;
  }, {}), t);
}
function Tt(e, i, t) {
  if (i ?? (i = JSON.stringify(e)), he[i]--, he[i] === 0) {
    delete he[i];
    const [, n] = Le[i];
    delete Le[i], n.length = 0;
  }
  se.delete(t);
}
const mn = x({
  name: "FormKitSchema",
  props: {
    schema: {
      type: [Array, Object],
      required: !0
    },
    data: {
      type: Object,
      default: () => ({})
    },
    library: {
      type: Object,
      default: () => ({})
    },
    memoKey: {
      type: String,
      required: !1
    }
  },
  setup(e, i) {
    var t;
    const n = at();
    let l = {};
    se.set(l, []);
    let s = Rt(e.library, e.schema, e.memoKey), r, o;
    dn || ye(() => e.schema, (d, _) => {
      var V;
      const J = l;
      l = {}, s = Rt(e.library, e.schema, e.memoKey), r = Ft(s, o, l), d === _ && ((V = n == null ? void 0 : n.proxy) === null || V === void 0 ? void 0 : V.$forceUpdate)(), Tt(e.schema, e.memoKey, J);
    }, { deep: !0 }), ae(() => {
      var d;
      o = Object.assign(yn((d = e.data) !== null && d !== void 0 ? d : {}), {
        slots: i.slots
      }), i.slots, r = Ft(s, o, l);
    });
    function u() {
      Tt(e.schema, e.memoKey, l), o.node && o.node.destroy(), o.slots = null, o = null, r = null;
    }
    return vn(u), El((t = at()) === null || t === void 0 ? void 0 : t.appContext.app, u), () => r ? r() : null;
  }
}), zl = {
  config: {
    type: Object,
    default: {}
  },
  classes: {
    type: Object,
    required: !1
  },
  delay: {
    type: Number,
    required: !1
  },
  dynamic: {
    type: Boolean,
    required: !1
  },
  errors: {
    type: Array,
    default: []
  },
  inputErrors: {
    type: Object,
    default: () => ({})
  },
  index: {
    type: Number,
    required: !1
  },
  id: {
    type: String,
    required: !1
  },
  modelValue: {
    required: !1
  },
  name: {
    type: String,
    required: !1
  },
  parent: {
    type: Object,
    required: !1
  },
  plugins: {
    type: Array,
    default: []
  },
  sectionsSchema: {
    type: Object,
    default: {}
  },
  sync: {
    type: Boolean,
    required: !1
  },
  type: {
    type: [String, Object],
    default: "text"
  },
  validation: {
    type: [String, Array],
    required: !1
  },
  validationMessages: {
    type: Object,
    required: !1
  },
  validationRules: {
    type: Object,
    required: !1
  },
  validationLabel: {
    type: [String, Function],
    required: !1
  }
}, Ul = zl, Kl = typeof window > "u", Ze = Symbol("FormKitParent");
x({
  props: Ul,
  emits: {
    /* eslint-disable @typescript-eslint/no-unused-vars */
    input: (e, i) => !0,
    inputRaw: (e, i) => !0,
    "update:modelValue": (e) => !0,
    node: (e) => !!e,
    submit: (e, i) => !0,
    submitRaw: (e, i) => !0,
    submitInvalid: (e) => !0
    /* eslint-enable @typescript-eslint/no-unused-vars */
  },
  inheritAttrs: !1,
  setup(e, i) {
    const t = Wl(e, i);
    if (t.props.definition || W(600, t), t.props.definition.component)
      return () => {
        var o;
        return Ae((o = t.props.definition) === null || o === void 0 ? void 0 : o.component, {
          context: t.context
        }, { ...i.slots });
      };
    const n = Me([]);
    let l = t.props.definition.schemaMemoKey;
    const s = () => {
      var o, u;
      const d = (u = (o = t.props) === null || o === void 0 ? void 0 : o.definition) === null || u === void 0 ? void 0 : u.schema;
      d || W(601, t), typeof d == "function" ? (n.value = d({ ...e.sectionsSchema }), (l && e.sectionsSchema || "memoKey" in d && typeof d.memoKey == "string") && (l = (l ?? (d == null ? void 0 : d.memoKey)) + JSON.stringify(e.sectionsSchema))) : n.value = d;
    };
    s(), Kl || t.on("schema", s), i.emit("node", t);
    const r = t.props.definition.library;
    return i.expose({ node: t }), () => Ae(mn, { schema: n.value, data: t.context, library: r, memoKey: l }, { ...i.slots });
  }
});
const bn = Symbol.for("FormKitOptions"), lt = [
  "help",
  "label",
  "ignore",
  "disabled",
  "preserve",
  /^preserve(-e|E)rrors/,
  /^[a-z]+(?:-visibility|Visibility|-behavior|Behavior)$/,
  /^[a-zA-Z-]+(?:-class|Class)$/,
  "prefixIcon",
  "suffixIcon",
  /^[a-zA-Z-]+(?:-icon|Icon)$/
];
function Dt(e, i) {
  i.classes && Object.keys(i.classes).forEach((t) => {
    typeof t == "string" && (e.props[`_${t}Class`] = i.classes[t], dt(i.classes[t]) && t === "inner" && Object.values(i.classes[t]));
  });
}
function ql(e) {
  return e ? ["Submit", "SubmitRaw", "SubmitInvalid"].reduce((t, n) => {
    const l = `on${n}`;
    return l in e && typeof e[l] == "function" && (t[l] = e[l]), t;
  }, {}) : {};
}
function Wl(e, i, t = {}) {
  var n;
  const l = Object.assign({}, Oe(bn) || {}, t), s = at(), r = ql(s == null ? void 0 : s.vnode.props), o = "modelValue" in ((n = s == null ? void 0 : s.vnode.props) !== null && n !== void 0 ? n : {});
  let u = !1;
  hn(() => {
    u = !0;
  });
  const d = e.modelValue !== void 0 ? e.modelValue : de(i.attrs.value);
  function _() {
    const b = {
      ...me(e),
      ...r,
      __slots: i.slots
    }, c = Ct(me(i.attrs), lt);
    c.key || (c.key = Je()), b.attrs = c;
    const g = St(me(i.attrs), lt);
    for (const L in g)
      b[Fe(L)] = g[L];
    const k = { props: {} };
    return Dt(k, e), Object.assign(b, k.props), typeof b.type != "string" && (b.definition = b.type, delete b.type), b;
  }
  const V = _(), J = V.ignore ? null : e.parent || Oe(Ze, null), y = Hi(Ye(l || {}, {
    name: e.name || void 0,
    value: d,
    parent: J,
    plugins: (l.plugins || []).concat(e.plugins),
    config: e.config,
    props: V,
    index: e.index,
    sync: e.sync || e.dynamic
  }, !1, !0));
  y.props.definition || W(600, y);
  const I = Me(new Set(y.props.definition.props || []));
  y.on("added-props", ({ payload: b }) => {
    Array.isArray(b) && b.forEach((c) => I.value.add(c));
  });
  const f = M(() => lt.concat([...I.value]).reduce((b, c) => (typeof c == "string" ? (b.push(Fe(c)), b.push(Ut(c))) : b.push(c), b), []));
  ae(() => Dt(y, e));
  const m = me(e);
  for (const b in m)
    ye(() => e[b], () => {
      e[b] !== void 0 && (y.props[b] = e[b]);
    });
  const p = /* @__PURE__ */ new Set(), h = me(i.attrs);
  ae(() => {
    $(St(h, f.value));
  });
  function $(b) {
    p.forEach((c) => {
      c(), p.delete(c);
    });
    for (const c in b) {
      const g = Fe(c);
      p.add(ye(() => i.attrs[c], () => {
        y.props[g] = i.attrs[c];
      }));
    }
  }
  if (ae(() => {
    const b = Ct(me(i.attrs), f.value);
    "multiple" in b && (b.multiple = Y(b.multiple)), typeof b.onBlur == "function" && (b.onBlur = jn(b.onBlur)), y.props.attrs = Object.assign({}, y.props.attrs || {}, b);
  }), ae(() => {
    const b = e.errors.map((c) => re({
      key: pt(c),
      type: "error",
      value: c,
      meta: { source: "prop" }
    }));
    y.store.apply(b, (c) => c.type === "error" && c.meta.source === "prop");
  }), y.type !== "input") {
    const b = `${y.name}-prop`;
    ae(() => {
      const c = Object.keys(e.inputErrors);
      c.length || y.clearErrors(!0, b);
      const g = c.reduce((k, L) => {
        let R = e.inputErrors[L];
        return typeof R == "string" && (R = [R]), Array.isArray(R) && (k[L] = R.map((C) => re({
          key: C,
          type: "error",
          value: C,
          meta: { source: b }
        }))), k;
      }, {});
      y.store.apply(g, (k) => k.type === "error" && k.meta.source === b);
    });
  }
  ae(() => Object.assign(y.config, e.config)), y.type !== "input" && _n(Ze, y);
  let v;
  return y.on("modelUpdated", () => {
    var b, c;
    i.emit("inputRaw", (b = y.context) === null || b === void 0 ? void 0 : b.value, y), u && i.emit("input", (c = y.context) === null || c === void 0 ? void 0 : c.value, y), o && y.context && (v = de(y.value), i.emit("update:modelValue", Pn(y.value)));
  }), o && (ye(gn(e, "modelValue"), (b) => {
    ue(v, b) || y.input(b, !1);
  }, { deep: !0 }), y.value !== d && y.emit("modelUpdated")), $n(() => y.destroy()), y;
}
let Nl = 1;
function Hl(e) {
  return typeof e == "function" && e.length === 2 || typeof e == "object" && !Array.isArray(e) && !("$el" in e) && !("$cmp" in e) && !("if" in e);
}
function D(e, i = {}) {
  const t = {
    type: "input",
    ...i
  };
  let n;
  if (Hl(e)) {
    const l = `SchemaComponent${Nl++}`;
    n = w("input", () => ({
      $cmp: l,
      props: {
        context: "$node.context"
      }
    })), t.library = { [l]: wn(e) };
  } else
    typeof e == "function" ? n = e : n = w("input", () => de(e));
  return t.schema = Il(n || "Schema undefined"), t.schemaMemoKey || (t.schemaMemoKey = `${Math.random()}`), t;
}
const Zl = w("messages", () => ({
  $el: "ul",
  if: "$fns.length($messages)"
})), Jl = w("message", () => ({
  $el: "li",
  for: ["message", "$messages"],
  attrs: {
    key: "$message.key",
    id: "$id + '-' + $message.key",
    "data-message-type": "$message.type"
  }
})), Yl = Zl(Jl("$message.value"));
x({
  props: {
    node: {
      type: Object,
      required: !1
    },
    sectionsSchema: {
      type: Object,
      default: {}
    },
    defaultPosition: {
      type: [String, Boolean],
      default: !1
    }
  },
  setup(e, i) {
    const t = M(() => e.node || Oe(Ze, void 0));
    ye(t, () => {
      var s;
      !((s = t.value) === null || s === void 0) && s.context && !Y(e.defaultPosition) && (t.value.context.defaultMessagePlacement = !1);
    }, { immediate: !0 });
    const n = Yl(e.sectionsSchema || {}), l = M(() => {
      var s, r, o, u, d, _;
      return {
        messages: ((r = (s = t.value) === null || s === void 0 ? void 0 : s.context) === null || r === void 0 ? void 0 : r.messages) || {},
        fns: ((u = (o = t.value) === null || o === void 0 ? void 0 : o.context) === null || u === void 0 ? void 0 : u.fns) || {},
        classes: ((_ = (d = t.value) === null || d === void 0 ? void 0 : d.context) === null || _ === void 0 ? void 0 : _.classes) || {}
      };
    });
    return () => {
      var s;
      return !((s = t.value) === null || s === void 0) && s.context ? Ae(mn, { schema: n, data: l.value }, { ...i.slots }) : null;
    };
  }
});
x({
  name: "FormKitIcon",
  props: {
    icon: {
      type: String,
      default: ""
    },
    iconLoader: {
      type: Function,
      default: null
    },
    iconLoaderUrl: {
      type: Function,
      default: null
    }
  },
  setup(e) {
    var i, t;
    const n = Me(void 0), l = Oe(bn, {}), s = Oe(Ze, null);
    let r;
    function o() {
      if (!r || typeof r != "function")
        return;
      const u = r(e.icon);
      u instanceof Promise ? u.then((d) => {
        n.value = d;
      }) : n.value = u;
    }
    if (e.iconLoader && typeof e.iconLoader == "function")
      r = nt(e.iconLoader);
    else if (s && (!((i = s.props) === null || i === void 0) && i.iconLoader))
      r = nt(s.props.iconLoader);
    else if (e.iconLoaderUrl && typeof e.iconLoaderUrl == "function")
      r = nt(r, e.iconLoaderUrl);
    else {
      const u = (t = l == null ? void 0 : l.plugins) === null || t === void 0 ? void 0 : t.find((d) => typeof d.iconHandler == "function");
      u && (r = u.iconHandler);
    }
    return ye(() => e.icon, () => {
      o();
    }, { immediate: !0 }), () => e.icon && n.value ? Ae("span", {
      class: "formkit-icon",
      innerHTML: n.value
    }) : null;
  }
});
const Ql = /* @__PURE__ */ x({
  __name: "PrimeCalendar",
  props: {
    context: Object
  },
  setup(e) {
    const i = e, t = i.context, n = t == null ? void 0 : t.attrs;
    function l(o) {
      t == null || t.node.input(t == null ? void 0 : t._value);
    }
    function s(o) {
      t == null || t.node.input(o);
    }
    const r = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (o, u) => {
      const d = E("Calendar");
      return A(), O("div", null, [
        T(d, {
          modelValue: a(t)._value,
          "onUpdate:modelValue": u[0] || (u[0] = (_) => a(t)._value = _),
          "input-id": i.context.id,
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          "input-style": a(n).style,
          "input-class": a(r),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          "date-format": a(n).dateFormat,
          "selection-mode": a(n).selectionMode ?? "single",
          inline: a(n).inline ?? !1,
          "show-other-months": a(n).showOtherMonths ?? !0,
          "select-other-months": a(n).selectOtherMonths ?? !1,
          icon: a(n).icon,
          "show-icon": a(t).showIcon,
          "previous-icon": a(n).previousIcon ?? "pi pi-chevron-left",
          "next-icon": a(n).nextIcon ?? "pi pi-chevron-right",
          "increment-icon": a(n).incrementIcon ?? "pi pi-chevron-up",
          "decrement-icon": a(n).decrementIcon ?? "pi pi-chevron-down",
          "number-of-months": a(n).numberOfMonths ?? 1,
          "responsive-options": a(n).responsiveOptions,
          view: a(n).view ?? "date",
          "touch-u-i": a(n).touchUI ?? !1,
          "min-date": a(n).minDate,
          "max-date": a(n).maxDate,
          "disabled-dates": a(n).disabledDates,
          "disabled-days": a(n).disabledDays,
          "max-date-count": a(n).maxDateCount,
          "show-on-focus": a(n).showOnFocus ?? !0,
          "auto-z-index": a(n).autoZIndex ?? !0,
          "base-z-index": a(n).baseZIndex ?? 0,
          "show-button-bar": a(n).showButtonBar ?? !1,
          "show-time": a(n).showTime ?? !1,
          "time-only": a(n).timeOnly ?? !1,
          "short-year-cutoff": a(n).shortYearCutoff ?? "+10",
          "hour-format": a(n).hourFormat ?? "24",
          "step-hour": a(n).stepHour ?? 1,
          "step-minute": a(n).stepMinute ?? 1,
          "step-second": a(n).stepSecond ?? 1,
          "show-seconds": a(n).showSeconds ?? !1,
          "hide-on-date-time-select": a(n).hideOnDateTimeSelect ?? !1,
          "hide-on-range-selection": a(n).hideOnRangeSelection ?? !1,
          "time-separator": a(n).timeSeparator ?? ":",
          "show-week": a(n).showWeek ?? !1,
          "manual-input": a(n).manualInput ?? !0,
          "append-to": a(n).appendTo ?? "body",
          "panel-style": a(n).panelStyle,
          "panel-class": a(n).panelClass,
          onDateSelect: s,
          onInput: l
        }, null, 8, ["modelValue", "input-id", "disabled", "readonly", "input-style", "input-class", "tabindex", "aria-label", "aria-labelledby", "date-format", "selection-mode", "inline", "show-other-months", "select-other-months", "icon", "show-icon", "previous-icon", "next-icon", "increment-icon", "decrement-icon", "number-of-months", "responsive-options", "view", "touch-u-i", "min-date", "max-date", "disabled-dates", "disabled-days", "max-date-count", "show-on-focus", "auto-z-index", "base-z-index", "show-button-bar", "show-time", "time-only", "short-year-cutoff", "hour-format", "step-hour", "step-minute", "step-second", "show-seconds", "hide-on-date-time-select", "hide-on-range-selection", "time-separator", "show-week", "manual-input", "append-to", "panel-style", "panel-class"])
      ]);
    };
  }
}), Xl = {
  key: 0,
  class: "formkit-prime-left"
}, Gl = {
  key: 1,
  class: "formkit-prime-right"
}, ea = /* @__PURE__ */ x({
  __name: "PrimeCheckbox",
  props: {
    context: Object
  },
  setup(e) {
    const i = e, t = i.context, n = t == null ? void 0 : t.attrs;
    function l(r) {
      var o;
      t == null || t.node.input((o = i.context) == null ? void 0 : o._value);
    }
    const s = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (r, o) => {
      const u = E("Checkbox");
      return A(), O("div", null, [
        a(t).attrs.labelLeft ? (A(), O("span", Xl, ce(a(t).attrs.labelLeft), 1)) : oe("", !0),
        T(u, {
          modelValue: a(t)._value,
          "onUpdate:modelValue": o[0] || (o[0] = (d) => a(t)._value = d),
          "input-id": a(t).id,
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          "input-style": a(n).style,
          "input-class": a(s),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          binary: a(n).binary ?? !0,
          "true-value": a(n).trueValue ?? void 0,
          "false-value": a(n).falseValue ?? void 0,
          onInput: l
        }, null, 8, ["modelValue", "input-id", "disabled", "readonly", "input-style", "input-class", "tabindex", "aria-label", "aria-labelledby", "binary", "true-value", "false-value"]),
        a(t).attrs.labelRight ? (A(), O("span", Gl, ce(a(t).attrs.labelRight), 1)) : oe("", !0)
      ]);
    };
  }
}), ta = /* @__PURE__ */ x({
  __name: "PrimeChips",
  props: {
    context: Object
  },
  setup(e) {
    const i = e, t = i.context, n = t == null ? void 0 : t.attrs;
    function l(r) {
      var o;
      t == null || t.node.input((o = i.context) == null ? void 0 : o._value);
    }
    const s = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (r, o) => {
      const u = E("Chips");
      return A(), O("div", null, [
        T(u, {
          modelValue: a(t)._value,
          "onUpdate:modelValue": o[0] || (o[0] = (d) => a(t)._value = d),
          "input-id": a(t).id,
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          "input-style": a(n).style,
          "input-class": a(s),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          "allow-duplicate": a(n).allowDuplicate ?? !0,
          "add-on-blur": a(n).addOnBlur ?? !1,
          max: a(n).max ?? void 0,
          placeholder: a(n).placeholder,
          separator: a(n).separator,
          onAdd: l,
          onRemove: l
        }, null, 8, ["modelValue", "input-id", "disabled", "readonly", "input-style", "input-class", "tabindex", "aria-label", "aria-labelledby", "allow-duplicate", "add-on-blur", "max", "placeholder", "separator"])
      ]);
    };
  }
}), na = /* @__PURE__ */ x({
  __name: "PrimeColorPicker",
  props: {
    context: Object
  },
  setup(e) {
    const i = e, t = i.context, n = t == null ? void 0 : t.attrs;
    function l(s) {
      var r;
      t == null || t.node.input((r = i.context) == null ? void 0 : r._value);
    }
    return (s, r) => {
      const o = E("ColorPicker");
      return A(), O("div", null, [
        T(o, {
          modelValue: a(t)._value,
          "onUpdate:modelValue": r[0] || (r[0] = (u) => a(t)._value = u),
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          style: pe(a(n).style),
          "panel-class": a(n).class,
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          "default-color": a(n).defaultColor ?? "ff0000",
          inline: a(n).inline ?? !1,
          format: a(n).format,
          onChange: l
        }, null, 8, ["modelValue", "disabled", "readonly", "style", "panel-class", "tabindex", "aria-label", "aria-labelledby", "default-color", "inline", "format"])
      ]);
    };
  }
}), ia = /* @__PURE__ */ x({
  __name: "PrimeDropdown",
  props: {
    context: Object
  },
  setup(e) {
    const t = e.context, n = t == null ? void 0 : t.attrs;
    function l(o) {
      t == null || t.handlers.blur(o.value);
    }
    function s(o) {
      t == null || t.node.input(o.value);
    }
    const r = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (o, u) => {
      const d = E("Dropdown");
      return A(), O("div", null, [
        T(d, {
          modelValue: a(t)._value,
          "onUpdate:modelValue": u[0] || (u[0] = (_) => a(t)._value = _),
          "input-id": a(t).id,
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          "input-style": a(r),
          "input-class": a(n).class,
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          options: a(n).options,
          "option-label": a(n).optionLabel ?? "label",
          "option-value": a(n).optionValue ?? "value",
          placeholder: a(n).placeholder,
          filter: a(n).filter ?? !1,
          "show-clear": a(n).showClear ?? !1,
          onChange: s,
          onBlur: l
        }, null, 8, ["modelValue", "input-id", "disabled", "readonly", "input-style", "input-class", "tabindex", "aria-label", "aria-labelledby", "options", "option-label", "option-value", "placeholder", "filter", "show-clear"])
      ]);
    };
  }
}), la = /* @__PURE__ */ x({
  __name: "PrimeEditor",
  props: {
    context: Object
  },
  setup(e) {
    const t = e.context, n = t == null ? void 0 : t.attrs;
    function l(o) {
      t == null || t.node.input(o.htmlValue);
    }
    function s(o) {
      o.range === null && (t == null || t.handlers.blur(o.htmlValue));
    }
    const r = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (o, u) => {
      const d = E("Editor");
      return A(), O("div", null, [
        T(d, {
          id: a(t).id,
          modelValue: a(t)._value,
          "onUpdate:modelValue": u[0] || (u[0] = (_) => a(t)._value = _),
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          "editor-style": a(n).style,
          class: U(a(r)),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          placeholder: a(n).placeholder,
          formats: a(n).formats,
          modules: a(n).modules,
          onTextChange: l,
          onSelectionChange: s
        }, null, 8, ["id", "modelValue", "disabled", "readonly", "editor-style", "class", "tabindex", "aria-label", "aria-labelledby", "placeholder", "formats", "modules"])
      ]);
    };
  }
}), aa = /* @__PURE__ */ x({
  __name: "PrimeInputMask",
  props: {
    context: Object
  },
  setup(e) {
    const i = e, t = i.context, n = t == null ? void 0 : t.attrs;
    function l(r) {
      var o, u;
      t == null || t.node.input((o = i.context) == null ? void 0 : o._value), t == null || t.handlers.blur((u = i.context) == null ? void 0 : u._value);
    }
    const s = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (r, o) => {
      const u = E("InputMask");
      return A(), O("div", null, [
        T(u, {
          id: a(t).id,
          modelValue: a(t)._value,
          "onUpdate:modelValue": o[0] || (o[0] = (d) => a(t)._value = d),
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          "editor-style": a(n).style,
          class: U(a(s)),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          placeholder: a(n).placeholder,
          mask: a(n).mask ?? void 0,
          "slot-char": a(n).slotChar ?? "_",
          "auto-clear": a(n).autoClear ?? !0,
          unmask: a(n).unmask ?? !1,
          onBlur: l
        }, null, 8, ["id", "modelValue", "disabled", "readonly", "editor-style", "class", "tabindex", "aria-label", "aria-labelledby", "placeholder", "mask", "slot-char", "auto-clear", "unmask"])
      ]);
    };
  }
}), sa = /* @__PURE__ */ x({
  __name: "PrimeInputNumber",
  props: {
    context: Object
  },
  setup(e) {
    const t = e.context, n = t == null ? void 0 : t.attrs;
    function l(o) {
      t == null || t.handlers.blur(o.value);
    }
    function s(o) {
      t == null || t.node.input(o.value);
    }
    const r = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (o, u) => {
      const d = E("InputNumber");
      return A(), O("div", null, [
        T(d, {
          modelValue: a(t)._value,
          "onUpdate:modelValue": u[0] || (u[0] = (_) => a(t)._value = _),
          "input-id": a(t).id,
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          "input-style": a(n).style,
          "input-class": a(r),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          placeholder: a(n).placeholder,
          "use-grouping": a(n).useGrouping ?? !0,
          "min-fraction-digits": a(n).minFractionDigits ?? void 0,
          "max-fraction-digits": a(n).maxFractionDigits ?? void 0,
          locale: a(n).locale ?? void 0,
          mode: a(n).mode ?? void 0,
          currency: a(n).currency ?? void 0,
          prefix: a(n).prefix ?? void 0,
          suffix: a(n).suffix ?? void 0,
          "show-buttons": a(n).showButtons ?? void 0,
          "button-layout": a(n).buttonLayout ?? "stacked",
          step: a(n).step ?? void 0,
          onInput: s,
          onBlur: l
        }, null, 8, ["modelValue", "input-id", "disabled", "readonly", "input-style", "input-class", "tabindex", "aria-label", "aria-labelledby", "placeholder", "use-grouping", "min-fraction-digits", "max-fraction-digits", "locale", "mode", "currency", "prefix", "suffix", "show-buttons", "button-layout", "step"])
      ]);
    };
  }
}), ra = {
  key: 0,
  class: "formkit-prime-left"
}, oa = {
  key: 1,
  class: "formkit-prime-right"
}, ua = /* @__PURE__ */ x({
  __name: "PrimeInputSwitch",
  props: {
    context: Object
  },
  setup(e) {
    const i = e, t = i.context, n = t == null ? void 0 : t.attrs;
    function l(r) {
      var o;
      t == null || t.node.input((o = i.context) == null ? void 0 : o._value);
    }
    const s = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (r, o) => {
      const u = E("InputSwitch");
      return A(), O("div", {
        class: U(a(n).option_class)
      }, [
        a(t).attrs.labelLeft ? (A(), O("span", ra, ce(a(t).attrs.labelLeft), 1)) : oe("", !0),
        T(u, {
          modelValue: a(t)._value,
          "onUpdate:modelValue": o[0] || (o[0] = (d) => a(t)._value = d),
          "input-id": a(t).id,
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          "input-style": a(n).style,
          "input-class": a(s),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          "true-value": a(n).trueValue ?? void 0,
          "false-value": a(n).falseValue ?? void 0,
          onInput: l
        }, null, 8, ["modelValue", "input-id", "disabled", "readonly", "input-style", "input-class", "tabindex", "aria-label", "aria-labelledby", "true-value", "false-value"]),
        a(t).attrs.labelRight ? (A(), O("span", oa, ce(a(t).attrs.labelRight), 1)) : oe("", !0)
      ], 2);
    };
  }
}), fa = /* @__PURE__ */ x({
  __name: "PrimeInputText",
  props: {
    context: Object
  },
  setup(e) {
    const t = e.context, n = t == null ? void 0 : t.attrs;
    function l() {
      return (t == null ? void 0 : t.iconLeft) && (t == null ? void 0 : t.iconLeft.length) > 0;
    }
    function s() {
      return (t == null ? void 0 : t.iconRight) && (t == null ? void 0 : t.iconRight.length) > 0;
    }
    function r() {
      let _ = "";
      return l() && (_ = `${_}p-input-icon-left `), s() && (_ = `${_}p-input-icon-right `), _;
    }
    function o(_) {
      t == null || t.handlers.blur(_.target.value);
    }
    function u(_) {
      t == null || t.node.input(_.target.value);
    }
    const d = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (_, V) => {
      const J = E("InputText");
      return A(), O("div", null, [
        Bt("span", {
          class: U(r())
        }, [
          l() ? (A(), O("i", {
            key: 0,
            class: U(a(t).iconLeft)
          }, null, 2)) : oe("", !0),
          T(J, {
            id: a(t).id,
            modelValue: a(t)._value,
            "onUpdate:modelValue": V[0] || (V[0] = (y) => a(t)._value = y),
            disabled: a(n)._disabled ?? !1,
            readonly: a(n)._readonly ?? !1,
            style: pe(a(n).style),
            class: U(a(d)),
            tabindex: a(n).tabindex,
            "aria-label": a(n).ariaLabel,
            "aria-labelledby": a(n).ariaLabelledby,
            placeholder: a(n).placeholder,
            onInput: u,
            onBlur: o
          }, null, 8, ["id", "modelValue", "disabled", "readonly", "style", "class", "tabindex", "aria-label", "aria-labelledby", "placeholder"]),
          s ? (A(), O("i", {
            key: 1,
            class: U(a(t).iconRight)
          }, null, 2)) : oe("", !0)
        ], 2)
      ]);
    };
  }
}), ca = /* @__PURE__ */ x({
  __name: "PrimeTextarea",
  props: {
    context: Object
  },
  setup(e) {
    const t = e.context, n = t == null ? void 0 : t.attrs;
    function l(o) {
      t == null || t.handlers.blur(o.target.value);
    }
    function s(o) {
      t == null || t.node.input(o.target.value);
    }
    const r = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (o, u) => {
      const d = E("Textarea");
      return A(), O("div", null, [
        T(d, {
          id: a(t).id,
          modelValue: a(t)._value,
          "onUpdate:modelValue": u[0] || (u[0] = (_) => a(t)._value = _),
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          style: pe(a(n).style),
          class: U(a(r)),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          placeholder: a(n).placeholder,
          rows: a(t).rows ?? 3,
          onInput: s,
          onBlur: l
        }, null, 8, ["id", "modelValue", "disabled", "readonly", "style", "class", "tabindex", "aria-label", "aria-labelledby", "placeholder", "rows"])
      ]);
    };
  }
}), da = /* @__PURE__ */ x({
  __name: "PrimeKnob",
  props: {
    context: Object
  },
  setup(e) {
    const t = e.context, n = t == null ? void 0 : t.attrs;
    function l(r) {
      t == null || t.node.input(r), t == null || t.handlers.blur(r);
    }
    const s = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (r, o) => {
      const u = E("Knob");
      return A(), O("div", null, [
        T(u, {
          id: a(t).id,
          modelValue: a(t)._value,
          "onUpdate:modelValue": o[0] || (o[0] = (d) => a(t)._value = d),
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          style: pe(a(n).style),
          class: U(a(s)),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          min: a(n).min ?? 0,
          max: a(n).max ?? 100,
          step: a(n).step ?? void 0,
          size: a(n).size ?? 100,
          "stroke-width": a(n).strokeWidth ?? 14,
          "show-value": a(n).showValue ?? !0,
          "value-color": a(n).valueColor ?? void 0,
          "range-color": a(n).rangeColor ?? void 0,
          "text-color": a(n).textColor ?? void 0,
          "value-template": a(n).valueTemplate ?? void 0,
          onChange: l
        }, null, 8, ["id", "modelValue", "disabled", "readonly", "style", "class", "tabindex", "aria-label", "aria-labelledby", "min", "max", "step", "size", "stroke-width", "show-value", "value-color", "range-color", "text-color", "value-template"])
      ]);
    };
  }
}), pa = /* @__PURE__ */ x({
  __name: "PrimeMultiSelect",
  props: {
    context: Object
  },
  setup(e) {
    const i = e, t = i.context, n = t == null ? void 0 : t.attrs;
    function l(r) {
      var o;
      t == null || t.node.input((o = i.context) == null ? void 0 : o._value);
    }
    const s = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (r, o) => {
      const u = E("MultiSelect");
      return A(), O("div", null, [
        T(u, {
          modelValue: a(t)._value,
          "onUpdate:modelValue": o[0] || (o[0] = (d) => a(t)._value = d),
          "input-id": a(t).id,
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          "list-style": a(n).style,
          class: U(a(s)),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          options: a(n).options,
          "option-label": a(n).optionLabel ?? "label",
          "option-value": a(n).optionValue ?? "value",
          filter: a(n).filter ?? !1,
          onChange: l
        }, null, 8, ["modelValue", "input-id", "disabled", "readonly", "list-style", "class", "tabindex", "aria-label", "aria-labelledby", "options", "option-label", "option-value", "filter"])
      ]);
    };
  }
}), ma = /* @__PURE__ */ x({
  __name: "PrimeListbox",
  props: {
    context: Object
  },
  setup(e) {
    const i = e, t = i.context, n = t == null ? void 0 : t.attrs;
    function l(r) {
      var o;
      t == null || t.node.input((o = i.context) == null ? void 0 : o._value);
    }
    const s = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (r, o) => {
      var d;
      const u = E("Listbox");
      return A(), O("div", null, [
        T(u, {
          id: a(t).id,
          modelValue: a(t)._value,
          "onUpdate:modelValue": o[0] || (o[0] = (_) => a(t)._value = _),
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          "list-style": a(n).style,
          class: U(a(s)),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          options: (d = a(n)) == null ? void 0 : d.options,
          "option-label": a(n).optionLabel ?? "label",
          "option-value": a(n).optionValue ?? "value",
          multiple: a(n).multiple ?? !1,
          filter: a(n).filter ?? !1,
          "filter-icon": a(n).filterIcon,
          "filter-placeholder": a(n).filterPlaceholder,
          "filter-locale": a(n).filterLocale,
          "filter-match-mode": a(n).filterMatchMode,
          "auto-option-focus": a(n).autoOptionFocus ?? !0,
          "select-on-focus": a(n).selectOnFocus ?? !1,
          onChange: l
        }, null, 8, ["id", "modelValue", "disabled", "readonly", "list-style", "class", "tabindex", "aria-label", "aria-labelledby", "options", "option-label", "option-value", "multiple", "filter", "filter-icon", "filter-placeholder", "filter-locale", "filter-match-mode", "auto-option-focus", "select-on-focus"])
      ]);
    };
  }
}), ba = /* @__PURE__ */ x({
  __name: "PrimePassword",
  props: {
    context: Object
  },
  setup(e) {
    const t = e.context, n = t == null ? void 0 : t.attrs;
    function l(o) {
      t == null || t.handlers.blur(o.target.value);
    }
    function s(o) {
      t == null || t.node.input(o.target.value);
    }
    const r = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (o, u) => {
      const d = E("Password");
      return A(), O("div", null, [
        T(d, {
          modelValue: a(t)._value,
          "onUpdate:modelValue": u[0] || (u[0] = (_) => a(t)._value = _),
          "input-id": a(t).id,
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          "input-style": a(n).style,
          "input-class": a(r),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          placeholder: a(n).placeholder,
          "medium-regex": a(n).mediumRegex ?? "^(((?=.*[a-z])(?=.*[A-Z]))|((?=.*[a-z])(?=.*[0-9]))|((?=.*[A-Z])(?=.*[0-9])))(?=.{6,})",
          "strong-regex": a(n).strongRegex ?? "^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{8,})",
          "prompt-label": a(n).promptLabel,
          "weak-label": a(n).weakLabel,
          "medium-label": a(n).mediumLabel,
          "strong-label": a(n).strongLabel,
          "hide-icon": a(n).hideIcon ?? "pi pi-eye-slash",
          "show-icon": a(n).showIcon ?? "pi pi-eye",
          feedback: a(t).feedback ?? !1,
          "toggle-mask": a(t).toggleMask ?? !1,
          onInput: s,
          onBlur: l
        }, null, 8, ["modelValue", "input-id", "disabled", "readonly", "input-style", "input-class", "tabindex", "aria-label", "aria-labelledby", "placeholder", "medium-regex", "strong-regex", "prompt-label", "weak-label", "medium-label", "strong-label", "hide-icon", "show-icon", "feedback", "toggle-mask"])
      ]);
    };
  }
}), ya = ["for"], va = /* @__PURE__ */ x({
  __name: "PrimeRadioButton",
  props: {
    context: Object
  },
  setup(e) {
    const i = e, t = i.context, n = t == null ? void 0 : t.attrs;
    function l(r) {
      var o;
      t == null || t.node.input((o = i.context) == null ? void 0 : o._value);
    }
    const s = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (r, o) => {
      const u = E("RadioButton");
      return A(), O("div", {
        class: U(a(n).options_class)
      }, [
        (A(!0), O(Sn, null, Vn(a(n).options, (d) => (A(), O("div", {
          key: d.value,
          class: U(a(n).option_class)
        }, [
          T(u, {
            modelValue: a(t)._value,
            "onUpdate:modelValue": o[0] || (o[0] = (_) => a(t)._value = _),
            name: a(n).name,
            value: d.value,
            "input-style": a(n).style,
            "input-class": a(s),
            onClick: l,
            onChange: l
          }, null, 8, ["modelValue", "name", "value", "input-style", "input-class"]),
          Bt("label", {
            for: d.value
          }, ce(d.label), 9, ya)
        ], 2))), 128))
      ], 2);
    };
  }
}), ha = /* @__PURE__ */ x({
  __name: "PrimeRating",
  props: {
    context: Object
  },
  setup(e) {
    const i = e, t = i.context, n = t == null ? void 0 : t.attrs;
    function l(r) {
      var o;
      t == null || t.node.input((o = i.context) == null ? void 0 : o._value);
    }
    const s = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (r, o) => {
      const u = E("Rating");
      return A(), O("div", null, [
        T(u, {
          id: a(t).id,
          modelValue: a(t)._value,
          "onUpdate:modelValue": o[0] || (o[0] = (d) => a(t)._value = d),
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          style: pe(a(n).style),
          class: U(a(s)),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          stars: a(n).stars ?? 5,
          cancel: a(n).cancel ?? !1,
          "on-icon": a(n).onIcon ?? "pi pi-star-fill",
          "off-icon": a(n).offIcon ?? "pi pi-star",
          "cancel-icon": a(n).cancelIcon ?? "pi pi-ban",
          onChange: l
        }, null, 8, ["id", "modelValue", "disabled", "readonly", "style", "class", "tabindex", "aria-label", "aria-labelledby", "stars", "cancel", "on-icon", "off-icon", "cancel-icon"])
      ]);
    };
  }
}), _a = /* @__PURE__ */ x({
  __name: "PrimeSlider",
  props: {
    context: Object
  },
  setup(e) {
    const t = e.context, n = t == null ? void 0 : t.attrs;
    function l(r) {
      t == null || t.node.input(r), t == null || t.handlers.blur(r);
    }
    const s = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (r, o) => {
      const u = E("Slider");
      return A(), O("div", null, [
        T(u, {
          id: a(t).id,
          modelValue: a(t)._value,
          "onUpdate:modelValue": o[0] || (o[0] = (d) => a(t)._value = d),
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          style: pe(a(n).style),
          class: U(a(s)),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          min: a(n).min ?? 0,
          max: a(n).max ?? 100,
          step: a(n).step ?? void 0,
          range: a(n).range ?? !1,
          orientation: a(n).orientation ?? "horizontal",
          onChange: l
        }, null, 8, ["id", "modelValue", "disabled", "readonly", "style", "class", "tabindex", "aria-label", "aria-labelledby", "min", "max", "step", "range", "orientation"])
      ]);
    };
  }
}), ga = /* @__PURE__ */ x({
  __name: "PrimeToggleButton",
  props: {
    context: Object
  },
  setup(e) {
    const i = e, t = i.context, n = t == null ? void 0 : t.attrs;
    function l(r) {
      var o;
      t == null || t.node.input((o = i.context) == null ? void 0 : o._value);
    }
    const s = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (r, o) => {
      const u = E("ToggleButton");
      return A(), O("div", null, [
        T(u, {
          modelValue: a(t)._value,
          "onUpdate:modelValue": o[0] || (o[0] = (d) => a(t)._value = d),
          "input-id": a(t).id,
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          "input-style": a(n).style,
          "input-class": a(s),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          "on-label": a(n).onLabel ?? "Yes",
          "off-label": a(n).offLabel ?? "No",
          "on-icon": a(n).onIcon ?? "pi pi-check",
          "off-icon": a(n).offIcon ?? "pi pi-times",
          "icon-pos": a(n).iconPos ?? "left",
          onChange: l
        }, null, 8, ["modelValue", "input-id", "disabled", "readonly", "input-style", "input-class", "tabindex", "aria-label", "aria-labelledby", "on-label", "off-label", "on-icon", "off-icon", "icon-pos"])
      ]);
    };
  }
}), $a = /* @__PURE__ */ x({
  __name: "PrimeSelectButton",
  props: {
    context: Object
  },
  setup(e) {
    const i = e, t = i.context, n = t == null ? void 0 : t.attrs;
    function l(r) {
      var o;
      t == null || t.node.input((o = i.context) == null ? void 0 : o._value);
    }
    const s = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (r, o) => {
      const u = E("SelectButton");
      return A(), O("div", null, [
        T(u, {
          id: a(t).id,
          modelValue: a(t)._value,
          "onUpdate:modelValue": o[0] || (o[0] = (d) => a(t)._value = d),
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          style: pe(a(n).style),
          class: U(a(s)),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          options: a(n).options,
          "option-label": a(n).optionLabel ?? "label",
          "option-value": a(n).optionValue ?? "value",
          "option-disabled": a(n).optionDisabled,
          multiple: a(n).multiple ?? !1,
          unselectable: a(n).unselectable ?? !0,
          "data-key": a(n).dataKey,
          onChange: l
        }, null, 8, ["id", "modelValue", "disabled", "readonly", "style", "class", "tabindex", "aria-label", "aria-labelledby", "options", "option-label", "option-value", "option-disabled", "multiple", "unselectable", "data-key"])
      ]);
    };
  }
}), wa = {
  key: 0,
  class: "formkit-prime-left"
}, ka = {
  key: 1,
  class: "formkit-prime-right"
}, Ca = /* @__PURE__ */ x({
  __name: "PrimeTriStateCheckbox",
  props: {
    context: Object
  },
  setup(e) {
    const i = e, t = i.context, n = t == null ? void 0 : t.attrs;
    function l(r) {
      var o;
      t == null || t.node.input((o = i.context) == null ? void 0 : o._value);
    }
    const s = M(() => t != null && t.state.validationVisible && !(t != null && t.state.valid) ? `${n == null ? void 0 : n.class} p-invalid` : n == null ? void 0 : n.class);
    return (r, o) => {
      const u = E("TriStateCheckbox");
      return A(), O("div", null, [
        a(t).attrs.labelLeft ? (A(), O("span", wa, ce(a(t).attrs.labelLeft), 1)) : oe("", !0),
        T(u, {
          modelValue: a(t)._value,
          "onUpdate:modelValue": o[0] || (o[0] = (d) => a(t)._value = d),
          "input-id": a(t).id,
          disabled: a(n)._disabled ?? !1,
          readonly: a(n)._readonly ?? !1,
          "input-style": a(n).style,
          "input-class": a(s),
          tabindex: a(n).tabindex,
          "aria-label": a(n).ariaLabel,
          "aria-labelledby": a(n).ariaLabelledby,
          onClick: l
        }, null, 8, ["modelValue", "input-id", "disabled", "readonly", "input-style", "input-class", "tabindex", "aria-label", "aria-labelledby"]),
        a(t).attrs.labelRight ? (A(), O("span", ka, ce(a(t).attrs.labelRight), 1)) : oe("", !0)
      ]);
    };
  }
}), Sa = D(fa, {
  props: ["iconRight", "iconLeft"]
}), Va = D(sa, {
  props: ["iconRight", "iconLeft"]
}), Ia = D(aa, {
  props: []
}), La = D(ba, {
  props: ["feedback", "toggleMask"]
}), Aa = D(ca, {
  props: ["rows"]
}), Oa = D(ea, {
  props: []
}), Pa = D(ua, {
  props: []
}), xa = D(la, {
  props: []
}), Ma = D(ia, {
  props: []
}), ja = D(pa, {
  props: []
}), Ea = D(ma, {
  props: []
}), Ra = D(Ql, {
  props: []
}), Fa = D(_a, {
  props: []
}), Ta = D(ha, {
  props: []
}), Da = D(va, {
  props: []
}), Ba = D(ta, {
  props: []
}), za = D(da, {
  props: []
}), Ua = D(na, {
  props: []
}), Ka = D(ga, {
  props: []
}), qa = D($a, {
  props: []
}), Wa = D(Ca, {
  props: []
}), Ha = {
  primeInputText: Sa,
  primeInputNumber: Va,
  primeInputMask: Ia,
  primePassword: La,
  primeCheckbox: Oa,
  primeInputSwitch: Pa,
  primeTextarea: Aa,
  primeEditor: xa,
  primeDropdown: Ma,
  primeMultiSelect: ja,
  primeCalendar: Ra,
  primeSlider: Fa,
  primeChips: Ba,
  primeKnob: za,
  primeRating: Ta,
  primeRadioButton: Da,
  primeColorPicker: Ua,
  primeToggleButton: Ka,
  primeListbox: Ea,
  primeSelectButton: qa,
  primeTriStateCheckbox: Wa
};
export {
  Ra as primeCalendarDefinition,
  Oa as primeCheckboxDefinition,
  Ba as primeChipsDefinition,
  Ua as primeColorPickerDefinition,
  Ma as primeDropdownDefinition,
  xa as primeEditorDefinition,
  Ia as primeInputMaskDefinition,
  Va as primeInputNumberDefinition,
  Pa as primeInputSwitchDefinition,
  Sa as primeInputTextDefinition,
  Ha as primeInputs,
  za as primeKnobDefinition,
  Ea as primeListboxDefinition,
  ja as primeMultiSelectDefinition,
  La as primePasswordDefinition,
  Da as primeRadioButtonDefinition,
  Ta as primeRatingDefinition,
  qa as primeSelectButtonDefinition,
  Fa as primeSliderDefinition,
  Aa as primeTextareaDefinition,
  Ka as primeToggleButtonDefinition,
  Wa as primeTriStateCheckboxDefinition
};
